---
title: "Combining_16S"
author: "Abby Sulesky"
date: "2023-06-07"
output:
  html_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, echo=FALSE, results='hide', message=FALSE, warning=FALSE}
library(microViz)
library(corncob)
library(ggraph)
library(DT)
library(phyloseq)
library(ggplot2)
library(tidyverse)
library(decontam)
library(scales)
library(vegan)
library(microbiome)
library(ComplexHeatmap)
library(dplyr)
library(patchwork)
library(pairwiseAdonis)
library(indicspecies)
library(VennDiagram)
```

# Update metadata in MSU combined phyloseq
```{r, echo=FALSE, results='hide'}

MSU_phyloseq <- readRDS("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/root_rhizo_control_drought_rarefiedphyloseq.rds")

OTU <- otu_table(MSU_phyloseq, taxa_are_rows = TRUE)
tax <- tax_table(MSU_phyloseq)
meta <- read.csv("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/combined_metadata_MSU.csv", row.names=1)
meta <- sample_data(meta)

MSU_phyloseq <- merge_phyloseq(OTU, meta, tax)
sample_data(MSU_phyloseq)

MSU_phyloseq

saveRDS(MSU_phyloseq, file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/updated_MSU_CD_phyloseq.rds", compress=TRUE)
```

# Update metadata in INRAE dataset
```{r, echo=FALSE, results='hide'}

INRAE_phyloseq <- readRDS("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/France/Fr_phyloseq.rds")

OTU <- otu_table(INRAE_phyloseq, taxa_are_rows = TRUE)
tax <- tax_table(INRAE_phyloseq)
meta <- read.csv("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/France/all_design_metadata_AFRI.csv", row.names=1)
meta <- sample_data(meta)

INRAE_phyloseq <- merge_phyloseq(OTU, meta, tax)

INRAE_root_rhizo_CD <- INRAE_phyloseq %>% 
  ps_filter(Compartment != "Mature_Seed", .keep_all_taxa = TRUE) %>%
  ps_filter(Compartment != "Single_Seed", .keep_all_taxa = TRUE) %>%
  ps_filter(Compartment != "Fresh_Seed", .keep_all_taxa = TRUE) %>%
  ps_filter(Compartment != "Leaf", .keep_all_taxa = TRUE) %>%
  ps_filter(Compartment != "Stem", .keep_all_taxa = TRUE) %>%
  ps_filter(Compartment != "Seedling", .keep_all_taxa = TRUE) %>%
  ps_filter(Treatment != "Nutrient", .keep_all_taxa = TRUE)

INRAE_root_rhizo_CD

sample_data(INRAE_root_rhizo_CD)

saveRDS(INRAE_root_rhizo_CD, file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/INRAE_root_rhizo_CD.rds", compress=TRUE)
#"INRAE_root_rhizo_CD.rds" is not rarefied data set

# Rarefying to 2500 reads
set.seed(333)
INRAE_root_rhizo_CD_rare <- rarefy_even_depth(INRAE_root_rhizo_CD, sample.size = 2500,
                                         rngseed = 333, replace = FALSE, trimOTUs = TRUE, verbose = TRUE)
# 10 samples removed, 153 ASVs were removed, 1116 ASVs and 110 samples remaining
```
# Combine into one phjyloseq object
```{r, echo=FALSE}

INRAE_root_rhizo_CD <- readRDS(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/INRAE_root_rhizo_CD_rare.rds") # "INRAE_root_rhizo_CD_rare.rds" is the rarefied dataset at 2500 reads
MSU_phyloseq <- readRDS(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/updated_MSU_CD_phyloseq.rds")

all_16S_phyloseq <- merge_phyloseq(MSU_phyloseq, INRAE_root_rhizo_CD)
sample_data(all_16S_phyloseq)

# subset G1 and G2
all_16S_gen1 <- all_16S_phyloseq %>% 
  ps_filter(Generation == "G1")

all_16S_gen2 <- all_16S_phyloseq %>% 
  ps_filter(Generation == "G2")

```

# Alpha diversity with Grids
This may not work because the data needs such drastically different scales. Going to separate roots and rhizosphere and try facet_wrap() instead
```{r}
# G1
G1_observed_alphaplot <- plot_richness(all_16S_gen1, x="Treatment", measures=c("Observed"), color = "Treatment") + geom_boxplot() + labs(title="Gen1 Alpha Diversity")

G1_observed_alphaplot + facet_grid(rows=vars(Compartment), cols=vars(Group), scales="free")

# G2
G2_observed_alphaplot <- plot_richness(all_16S_gen2, x="G1_G2", measures=c("Observed"), color = "G1_G2") + geom_violin() + geom_jitter(width=0.2) + labs(title="Gen2 Alpha Diversity")

G2_observed_alphaplot + facet_grid(rows=vars(Compartment), cols=vars(Group), scales="free")
G2_observed_alphaplot + facet_wrap(~Group + Compartment, scales="free")
```


# Separate root and rhizo and do facet_wrap()
```{r}
# Subset
root_16S_gen1 <- all_16S_gen1 %>% 
  ps_filter(Compartment == "Root")

root_16S_gen2 <- all_16S_gen2 %>% 
  ps_filter(Compartment == "Root")

rhizo_16S_gen1 <- all_16S_gen1 %>% 
  ps_filter(Compartment == "Rhizosphere")

rhizo_16S_gen2 <- all_16S_gen2 %>% 
  ps_filter(Compartment == "Rhizosphere")

#sample_data(root_16S_gen2)

# code for background theme for ggplot so I don't have to have it in every plot
my_theme <- theme(panel.background = element_rect(fill = "white", colour = "white"), 
             panel.grid.major = element_line(size = 0.25, linetype = 'solid', colour = "light gray"),
             panel.grid.minor = element_line(size = 0.25, linetype = 'solid', colour = "light gray"))


# G1 Root
G1_observed_alphaplot_root <- plot_richness(root_16S_gen1, x="Treatment", measures=c("Observed"), color = "Treatment") + geom_boxplot() + labs(title="Gen1 Root 16S Alpha Diversity") + scale_color_manual(name = "Treatment", values = c("#3399FF","#FFCC00"), labels = c("Control", "Drought"), guide = guide_legend(override.aes = list(size = 3), alpha = 1)) + my_theme

G1_observed_alphaplot_root + facet_wrap(vars(Group),nrow=1, scales="free")
p1 <- G1_observed_alphaplot_root + facet_wrap(vars(Group),nrow=1, scales="free")

# G1 Rhizosphere
G1_observed_alphaplot_rhizo <- plot_richness(rhizo_16S_gen1, x="Treatment", measures=c("Observed"), color = "Treatment") + geom_boxplot() + labs(title="Gen1 Rhizosphere 16S Alpha Diversity") + scale_color_manual(name = "Treatment", values = c("#3399FF","#FFCC00"), labels = c("Control", "Drought"), guide = guide_legend(override.aes = list(size = 3), alpha = 1)) + my_theme

G1_observed_alphaplot_rhizo + facet_wrap(vars(Group),nrow=1, scales="free")
p2 <- G1_observed_alphaplot_rhizo + facet_wrap(vars(Group),nrow=1, scales="free")


G1_alpha <- (p1/p2) +
  plot_annotation(tag_levels = 'A')
G1_alpha


# G2 Root
G2_observed_alphaplot_root <- plot_richness(root_16S_gen2, x="G1_G2", measures=c("Observed"), color = "G1_G2") + geom_violin() + geom_jitter(width=0.2) + labs(title="Gen2 Root 16S Alpha Diversity") + scale_color_manual(name = "G1_G2 Treatments", values = c("#3399FF","#99CCFF", "#FFCC00", "#FF6600"), labels = c("Control_Control", "Control_Drought", "Drought_Control", "Drought_Drought")) + my_theme

G2_observed_alphaplot_root + facet_wrap(vars(Group),nrow=1, scales="free")

# G2 Rhizo
G2_observed_alphaplot_rhizo <- plot_richness(rhizo_16S_gen2, x="G1_G2", measures=c("Observed"), color = "G1_G2") + geom_violin() + geom_jitter(width=0.2) + labs(title="Gen2 Rhizosphere 16S Alpha Diversity") + scale_color_manual(name = "G1_G2 Treatments", values = c("#3399FF","#99CCFF", "#FFCC00", "#FF6600"), labels = c("Control_Control", "Control_Drought", "Drought_Control", "Drought_Drought")) + my_theme

G2_observed_alphaplot_rhizo + facet_wrap(vars(Group),nrow=1, scales="free")

```

# New alpha diversity keeping MSU and INRAE separate
### Gen 1
```{r, warning=FALSE}
# Subset INRAE G1 root and rhizo
INRAE_gen1 <- INRAE_root_rhizo_CD %>%
  ps_filter(Generation == "G1")

INRAE_gen1_root <- INRAE_gen1 %>% 
  ps_filter(Compartment == "Root")

INRAE_gen1_rhizo <- INRAE_gen1 %>% 
  ps_filter(Compartment == "Rhizosphere")

# inrae G1 root and rhizo alpha figures:
inrae_root_g1_alpha <- plot_richness(INRAE_gen1_root, x="Treatment", measures=c("Observed"), 
                                     color = "Treatment") + 
  geom_boxplot() + 
  eom_point(size=3, shape=21, fill="white")+
  scale_color_manual(name = "Treatment", values = c("#3399FF","#FFCC00"), labels = c("Control", "Drought"), guide =   guide_legend(override.aes = list(size = 3), alpha = 1)) + 
  scale_y_continuous(limits = c(40, 200))+
  labs(title="G1 Root Alpha Diversity (FR)") + 
  ylab("ASVs per Root Sample") +
  my_theme + theme(legend.position="none", axis.text.x=element_blank(), 
        axis.title.x=element_blank(), 
        plot.tag = element_text(face="bold")) +
  facet_wrap(vars(Genotype),nrow=1, scales="free")
 # There are no significant pairwise comparison between Drought vs Control within Genotype
 # geom_text(inherit.aes=FALSE, data = inrae_rootG1_anno, aes(x = xstar,  y = ystar, label = lab), size=6) +
 # geom_segment(inherit.aes=FALSE, data = inrae_rootG1_anno, aes(x = x1, xend = x1, y = y1, yend = y2), colour = "black") +
 # geom_segment(inherit.aes=FALSE, data = inrae_rootG1_anno, aes(x = x2, xend = x2, y = y1, yend = y2), colour = "black") +
 # geom_segment(inherit.aes=FALSE, data = inrae_rootG1_anno, aes(x = x1, xend = x2, y = y2, yend = y2), colour = "black")

inrae_root_g1_alpha

#inrae_rootG1_anno <- data.frame(x1 = c(1), x2 = c(2), 
                 #  y1 = c(243), y2 = c(248), 
                 #  xstar = c(1.5), ystar = c(251),
                 # lab = c("*"),
                  # Genotype = c("Flavert"))
# stats
inrae_root_G1alpha <- estimate_richness(INRAE_gen1_root, measures=c("Observed"))
inrae_root_G1alpha <- cbind(rownames(inrae_root_G1alpha), inrae_root_G1alpha) 
colnames(inrae_root_G1alpha)[1] <- "SampleID"
inrae_root_G1alpha

write.csv(inrae_root_G1alpha, file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/inrae_root_G1alpha.csv")

inrae_root_G1alpha <- read.csv(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/inrae_root_G1alpha_fix.csv")

iG1root_metadata <- data.frame(sample_data(INRAE_gen1_root))

iG1root_data <- left_join(iG1root_metadata, inrae_root_G1alpha, by="SampleID")

iG1root_stats <- aov(Observed ~ Genotype*G1_treatment, data=iG1root_data) 
summary(iG1root_stats)
# Genotype 1   7031    7031   6.951 0.0180 *
iG1root_tukey <- tukey_hsd(iG1root_data, Observed ~ Genotype*G1_treatment)
iG1root_tukey

# Flavert vs. RH ***
# Flavert control vs. drought NS
# RH control vs. drought NS

inrae_rhizo_g1_alpha <- plot_richness(INRAE_gen1_rhizo, x="Treatment", measures=c("Observed"), 
                                     color = "Treatment") + 
  geom_boxplot() + 
  geom_point(size=3, shape=21, fill="white")+
  labs(title="G1 Rhizosphere Alpha Diversity (FR)") + 
  ylab("ASVs per Rhizosphere Sample") +
  scale_color_manual(name = "Treatment", values = c("#3399FF","#FFCC00"), labels = c("Control", "Drought"), guide = guide_legend(override.aes = list(size = 3), alpha = 1)) + my_theme + facet_wrap(vars(Genotype),nrow=1, scales="free") + theme(legend.position="none", plot.tag = element_text(face="bold"))

inrae_rhizo_g1_alpha

# stats
inrae_rhizo_G1alpha <- estimate_richness(INRAE_gen1_rhizo, measures=c("Observed"))
inrae_rhizo_G1alpha <- cbind(rownames(inrae_rhizo_G1alpha), inrae_rhizo_G1alpha) 
colnames(inrae_rhizo_G1alpha)[1] <- "SampleID"
inrae_rhizo_G1alpha

write.csv(inrae_rhizo_G1alpha, file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/inrae_rhizo_G1alpha.csv")

inrae_rhizo_G1alpha <- read.csv(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/inrae_rhizo_G1alpha_fix.csv")

iG1rhizo_metadata <- data.frame(sample_data(INRAE_gen1_rhizo))

iG1rhizo_data <- left_join(iG1rhizo_metadata, inrae_rhizo_G1alpha, by="SampleID")

iG1rhizo_stats <- aov(Observed ~ Genotype*G1_treatment, data=iG1rhizo_data)
summary(iG1rhizo_stats)
#Genotype 1  17582   17582   4.866 0.0424 *
#Genotype:G1_treatment  1  17111   17111   4.735 0.0449 *
iG1rhizo_tukey <- tukey_hsd(iG1rhizo_data, Observed ~ Genotype*G1_treatment)
iG1rhizo_tukey

# Flavert vs. RH *
# Flavert control vs. drought NS
# RH control vs. drought NS
# Fl:drought vs. RH:drought *


# Subset MSU G1 root and rhizo
MSU_gen1 <- MSU_phyloseq %>%
  ps_filter(Generation == "G1")

MSU_gen1_root <- MSU_gen1 %>% 
  ps_filter(Compartment == "Root")

MSU_gen1_rhizo <- MSU_gen1 %>% 
  ps_filter(Compartment == "Rhizosphere")

# inrae G1 root and rhizo alpha figures:
MSU_root_g1_alpha <- plot_richness(MSU_gen1_root, x="Treatment", measures=c("Observed"), 
                                     color = "Treatment") + 
  geom_boxplot() + 
  geom_point(size=3, shape=21, fill="white")+
  labs(title="(USA)") +
  scale_color_manual(name = "Treatment", values = c("#3399FF","#FFCC00"), labels = c("Control", "Drought"), guide = guide_legend(override.aes = list(size = 3), alpha = 1)) + my_theme + theme(axis.text.x=element_blank(), axis.title.x=element_blank(), axis.title.y=element_blank(), plot.tag = element_text(face="bold")) + facet_wrap(~Genotype)

MSU_root_g1_alpha

# stats
msu_root_G1alpha <- estimate_richness(MSU_gen1_root, measures=c("Observed"))
msu_root_G1alpha <- cbind(rownames(msu_root_G1alpha), msu_root_G1alpha) 
colnames(msu_root_G1alpha)[1] <- "SampleID"
msu_root_G1alpha

mG1root_metadata <- data.frame(sample_data(MSU_gen1_root))

mG1root_data <- left_join(mG1root_metadata, msu_root_G1alpha, by="SampleID")

mG1root_stats <- t.test(Observed ~ G1_treatment, data=mG1root_data)
mG1root_stats
# NS

MSU_rhizo_g1_alpha <- plot_richness(MSU_gen1_rhizo, x="Treatment", measures=c("Observed"), 
                                     color = "Treatment") + 
  geom_boxplot() + 
  geom_point(size=3, shape=21, fill="white")+
  labs(title="(USA)") +
  scale_color_manual(name = "Treatment", values = c("#3399FF","#FFCC00"), labels = c("Control", "Drought"), guide = guide_legend(override.aes = list(size = 3), alpha = 1)) + my_theme + theme(legend.position="none", axis.title.y=element_blank(), plot.tag = element_text(face="bold")) + facet_wrap(~Genotype)

MSU_rhizo_g1_alpha

# stats
msu_rhizo_G1alpha <- estimate_richness(MSU_gen1_rhizo, measures=c("Observed"))
msu_rhizo_G1alpha <- cbind(rownames(msu_rhizo_G1alpha), msu_rhizo_G1alpha) 
colnames(msu_rhizo_G1alpha)[1] <- "SampleID"
msu_rhizo_G1alpha

mG1rhizo_metadata <- data.frame(sample_data(MSU_gen1_rhizo))

mG1rhizo_data <- left_join(mG1rhizo_metadata, msu_rhizo_G1alpha, by="SampleID")

mG1rhizo_stats <- t.test(Observed ~ G1_treatment, data=mG1rhizo_data)
mG1rhizo_stats
# NS

gen1_full_alpha <- (inrae_root_g1_alpha + MSU_root_g1_alpha + plot_layout(ncol=2,widths=c(2,1))) / (inrae_rhizo_g1_alpha + MSU_rhizo_g1_alpha + plot_layout(ncol=2,widths=c(2,1))) + plot_annotation(tag_levels = 'A')
gen1_full_alpha


ggsave(plot=gen1_full_alpha, "/Users/Abby/OneDrive - Michigan State University/multigen_analysis/G1_alpha_figure.png", width=8, height=6, units="in")

### Fig. 4

ggsave(plot=gen1_full_alpha, "/PATH/Fig.4.tif",
       width=8, height=6,
       device = "tiff",
       units= "in", dpi = 400,
       compression="lzw", bg= "white")
```

### Gen2
```{r}
# Subset INRAE G1 root and rhizo
INRAE_gen2 <- INRAE_root_rhizo_CD %>%
  ps_filter(Generation == "G2")

INRAE_gen2_root <- INRAE_gen2 %>% 
  ps_filter(Compartment == "Root")

INRAE_gen2_rhizo <- INRAE_gen2 %>% 
  ps_filter(Compartment == "Rhizosphere")

# inrae G2 figures

a_iG2root <- data.frame(estimate_richness(INRAE_gen2_root, measures = "Observed"))
a_iG2root <- cbind(rownames(a_iG2root), a_iG2root) 
colnames(a_iG2root)[1] <- "SampleID"
a_iG2root
write.csv(a_iG2root, file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/a_iG2root.csv")
a_iG2root <- read.csv(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/a_iG2root_fix.csv")
meta_iG2root <- data.frame(sample_data(INRAE_gen2_root))
data_iG2root <- left_join(meta_iG2root, a_iG2root)

iG2root_alphaplot <- ggplot(data=data_iG2root, aes(x=G1_treatment, y=Observed, group=G1_G2, color = G1_G2)) + 
  geom_boxplot(position = position_dodge(width = 0.8)) + 
  geom_point(position=position_dodge(width=0.8),aes(group=G1_G2, col=G1_G2),size=3, shape=21, fill="white")+
  labs(title="G2 Root Alpha Diversity (FR)") + 
  scale_color_manual(name = "G1_G2 Consecutive\nTreatment", 
                     values = c("#3399FF","#99CCFF", "#FFCC00", "#FF6600"), 
                     labels = c("Control_Control", "Control_Drought", "Drought_Control", "Drought_Drought")) + ylab("ASVs per Root Sample") + 
  theme(legend.position="none", axis.text.x=element_blank(), 
        axis.title.x=element_blank(), 
        plot.tag = element_text(face="bold")) +
  my_theme + facet_wrap(~Genotype, scales="free")

iG2root_alphaplot 

aov_iG2root <- aov(Observed ~ Genotype*G1_treatment*G2_treatment, data=data_iG2root)
summary(aov_iG2root)
# #G2_treatment 1  309.0  308.96   4.312 0.0479 *
iG2root_tukey <- tukey_hsd(data_iG2root, Observed ~ Genotype*G1_treatment*G2_treatment)
iG2root_tukey

# inrae G2 figures

a_iG2rhizo <- data.frame(estimate_richness(INRAE_gen2_rhizo, measures = "Observed"))
a_iG2rhizo <- cbind(rownames(a_iG2rhizo), a_iG2rhizo) 
colnames(a_iG2rhizo)[1] <- "SampleID"
a_iG2rhizo
write.csv(a_iG2rhizo, file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/a_iG2rhizo.csv")
a_iG2rhizo <- read.csv(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/a_iG2rhizo_fix.csv")
meta_iG2rhizo <- data.frame(sample_data(INRAE_gen2_rhizo))
data_iG2rhizo <- left_join(meta_iG2rhizo, a_iG2rhizo)

iG2rhizo_anno <- data.frame(x1 = c(0.75, 0.75), x2 = c(1.75, 2.25), 
                            y1 = c(315, 330), y2 = c(320, 335), 
                            xstar = c(1.25, 1.5), ystar = c(322, 340),
                   lab = c("****", "****"),
                   Genotype = c("Flavert", "Flavert"))

iG2rhizo_alphaplot <- ggplot(data=data_iG2rhizo, aes(x=G1_treatment, y=Observed, group=G1_G2, color = G1_G2)) + 
  geom_boxplot(position = position_dodge(width = 0.8)) + 
  geom_point(position=position_dodge(width=0.8),aes(group=G1_G2, col=G1_G2),size=3, shape=21, fill="white")+
  ylab("ASVs per Rhizosphere Sample") +
  labs(title="G2 Rhizosphere Alpha Diversity (FR)") + 
  scale_y_continuous(limits = c(0, 350))+
  scale_color_manual(name = "G1_G2 Consecutive\nTreatment", 
                     values = c("#3399FF","#99CCFF", "#FFCC00", "#FF6600"), 
                     labels = c("Control_Control", "Control_Drought", "Drought_Control", "Drought_Drought")) + theme(legend.position="none", plot.tag = element_text(face="bold")) +
  my_theme + facet_wrap(~Genotype, scales="free") + 
  geom_text(inherit.aes=FALSE, data = iG2rhizo_anno, aes(x = xstar,  y = ystar, label = lab), size=6) +
  geom_segment(inherit.aes=FALSE, data = iG2rhizo_anno, aes(x = x1, xend = x1, 
           y = y1, yend = y2),
           colour = "black") +
  geom_segment(inherit.aes=FALSE, data = iG2rhizo_anno, aes(x = x2, xend = x2, 
           y = y1, yend = y2),
           colour = "black") +
  geom_segment(inherit.aes=FALSE, data = iG2rhizo_anno, aes(x = x1, xend = x2, 
           y = y2, yend = y2),
           colour = "black")

iG2rhizo_alphaplot 


aov_iG2rhizo <- aov(Observed ~ Genotype*G1_treatment*G2_treatment, data=data_iG2rhizo)
summary(aov_iG2rhizo)
#Genotype                            1  33489   33489  17.708  0.00024 ***
#G1_treatment                        1  47892   47892  25.325 2.54e-05 ***
#Genotype:G1_treatment               1  31299   31299  16.550  0.00035 ***
iG2rhizo_tukey <- tukey_hsd(data_iG2rhizo, Observed ~ Genotype*G1_treatment*G2_treatment)
iG2rhizo_tukey
# Fl vs. RH  ***
# G1_treatment  Control vs Drought 0.0000259  ****
# Fl G1 C vs.  Fl G1 D ***
# Fl C_C - D_C ****
# Fl C_C - D_D ****

iG2root_alphaplot
iG2rhizo_alphaplot


# MSU G2 alpha diversity
MSU_gen2 <- MSU_phyloseq %>%
  ps_filter(Generation == "G2")

MSU_gen2_root <- MSU_gen2 %>% 
  ps_filter(Compartment == "Root")

MSU_gen2_rhizo <- MSU_gen2 %>% 
  ps_filter(Compartment == "Rhizosphere")

# Roots
a_mG2root <- data.frame(estimate_richness(MSU_gen2_root, measures = "Observed"))
a_mG2root <- cbind(rownames(a_mG2root), a_mG2root) 
colnames(a_mG2root)[1] <- "SampleID"
a_mG2root
meta_mG2root <- data.frame(sample_data(MSU_gen2_root))
data_mG2root <- left_join(meta_mG2root, a_mG2root)

mG2root_alphaplot <- ggplot(data=data_mG2root, aes(x=G1_treatment, y=Observed, group=G1_G2, color = G1_G2)) + 
  geom_boxplot(position = position_dodge(width = 0.8)) +
  geom_point(position=position_dodge(width=0.8),aes(group=G1_G2, col=G1_G2),size=3, shape=21, fill="white")+
  labs(title="(USA)") + 
  scale_y_continuous(limits = c(300, 1600))+
  scale_color_manual(name = "G1_G2 Consecutive\nTreatment", 
                     values = c("#3399FF","#99CCFF", "#FFCC00", "#FF6600"), 
                     labels = c("Control_Control", "Control_Drought", "Drought_Control", "Drought_Drought")) + ylab("ASVs per Root Sample") + 
  theme(axis.text.x=element_blank(), 
        axis.title.x=element_blank(), axis.title.y=element_blank(), 
        plot.tag = element_text(face="bold")) +
  my_theme + facet_wrap(~Genotype)

mG2root_alphaplot 

aov_mG2root <- aov(Observed ~ G1_treatment*G2_treatment, data=data_mG2root)
summary(aov_mG2root)
# NS
mG2root_tukey <- tukey_hsd(data_mG2root, Observed ~ G1_treatment*G2_treatment)
mG2root_tukey


# Rhizosphere
a_mG2rhizo <- data.frame(estimate_richness(MSU_gen2_rhizo, measures = "Observed"))
a_mG2rhizo <- cbind(rownames(a_mG2rhizo), a_mG2rhizo) 
colnames(a_mG2rhizo)[1] <- "SampleID"
a_mG2rhizo
meta_mG2rhizo <- data.frame(sample_data(MSU_gen2_rhizo))
data_mG2rhizo <- left_join(meta_mG2rhizo, a_mG2rhizo)

mG2rhizo_alphaplot <- ggplot(data=data_mG2rhizo, aes(x=G1_treatment, y=Observed, group=G1_G2, color = G1_G2)) + 
  geom_boxplot(position = position_dodge(width = 0.8)) +
  geom_point(position=position_dodge(width=0.8),aes(group=G1_G2, col=G1_G2),size=3, shape=21, fill="white")+
  labs(title="(USA)") + 
  scale_y_continuous(limits = c(500, 2550))+
  scale_color_manual(name = "G1_G2 Consecutive\nTreatment", 
                     values = c("#3399FF","#99CCFF", "#FFCC00", "#FF6600"), 
                     labels = c("Control_Control", "Control_Drought", "Drought_Control", "Drought_Drought")) + ylab("ASVs per Root Sample") + 
  theme(legend.position="none", axis.title.y=element_blank(), 
        plot.tag = element_text(face="bold")) +
  my_theme + facet_wrap(~Genotype)

mG2rhizo_alphaplot 

aov_mG2rhizo <- aov(Observed ~ G1_treatment*G2_treatment, data=data_mG2rhizo)
summary(aov_mG2rhizo)
# interaction between G1 and G2 * but nothing significant between groups in post hoc
mG2rhizo_tukey <- tukey_hsd(data_mG2rhizo, Observed ~ G1_treatment*G2_treatment)
mG2rhizo_tukey


gen2_full_alpha <- (iG2root_alphaplot + mG2root_alphaplot + plot_layout(ncol=2,widths=c(2,1))) / (iG2rhizo_alphaplot + mG2rhizo_alphaplot + plot_layout(ncol=2,widths=c(2,1))) + plot_annotation(tag_levels = 'A')
gen2_full_alpha


ggsave(plot=gen2_full_alpha, "/Users/Abby/OneDrive - Michigan State University/multigen_analysis/G2_alpha_figure.png", width=11, height=8, units="in")

## Fig. 5

ggsave(plot=gen2_full_alpha, "/PATH/Fig.5.tif",
       width=11, height=8,
       device = "tiff",
       units= "in", dpi = 400,
       compression="lzw", bg= "white")
```


# Beta diversity INRAE
```{r}
# read the rarefied data set
INRAE_root_rhizo_CD <- readRDS(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/INRAE_root_rhizo_CD_rare.rds")

INRAE_gen1 <- INRAE_root_rhizo_CD %>%
  ps_filter(Generation == "G1")

INRAE_gen1_root <- INRAE_gen1 %>% 
  ps_filter(Compartment == "Root")

INRAE_gen1_rhizo <- INRAE_gen1 %>% 
  ps_filter(Compartment == "Rhizosphere")

INRAE_gen2 <- INRAE_root_rhizo_CD %>%
  ps_filter(Generation == "G2")

INRAE_gen2_root <- INRAE_gen2 %>% 
  ps_filter(Compartment == "Root")

INRAE_gen2_rhizo <- INRAE_gen2 %>% 
  ps_filter(Compartment == "Rhizosphere")

# need to separate genotypes

INRAE_gen1_rootF <- INRAE_gen1_root %>% 
  ps_filter(Genotype == "Flavert")

INRAE_gen1_rhizoF <- INRAE_gen1_rhizo %>% 
  ps_filter(Genotype == "Flavert")

INRAE_gen2_rootF <- INRAE_gen2_root %>% 
  ps_filter(Genotype == "Flavert")

INRAE_gen2_rhizoF <- INRAE_gen2_rhizo %>% 
  ps_filter(Genotype == "Flavert")

INRAE_gen1_rootRH <- INRAE_gen1_root %>% 
  ps_filter(Genotype == "Red_Hawk")

INRAE_gen1_rhizoRH <- INRAE_gen1_rhizo %>% 
  ps_filter(Genotype == "Red_Hawk")

INRAE_gen2_rootRH <- INRAE_gen2_root %>% 
  ps_filter(Genotype == "Red_Hawk")

INRAE_gen2_rhizoRH <- INRAE_gen2_rhizo %>% 
  ps_filter(Genotype == "Red_Hawk")


# Generation 1 roots
perm_iG1root <- how(nperm = 9999) 
meta_iG1root = data.frame(sample_data(INRAE_gen1_root))

iG1root_BCdist = vegdist(otu_table(INRAE_gen1_root), method = "bray", binary = FALSE)

dim(iG1root_BCdist)

set.seed(8)
iG1root_BC_adonis <- adonis2(formula = iG1root_BCdist ~ Genotype*G1_treatment, permutations = perm_iG1root, data = meta_iG1root)

iG1root_BC_adonis
# Genotype ***

# Flavert G1 root
perm_iG1rootF <- how(nperm = 9999) 
meta_iG1rootF = data.frame(sample_data(INRAE_gen1_rootF))

iG1rootF_BCdist = vegdist(otu_table(INRAE_gen1_rootF), method = "bray", binary = FALSE)

set.seed(8)
iG1rootF_BC_adonis <- adonis2(formula = iG1rootF_BCdist ~ G1_treatment, permutations = perm_iG1rootF, data = meta_iG1rootF)

iG1rootF_BC_adonis
# NS

# RedHawk G1 root
perm_iG1rootRH <- how(nperm = 9999) 
meta_iG1rootRH = data.frame(sample_data(INRAE_gen1_rootRH))

iG1rootRH_BCdist = vegdist(otu_table(INRAE_gen1_rootRH), method = "bray", binary = FALSE)

set.seed(8)
iG1rootRH_BC_adonis <- adonis2(formula = iG1rootRH_BCdist ~ G1_treatment, permutations = perm_iG1rootRH, data = meta_iG1rootRH)

iG1rootRH_BC_adonis
# NS


# Flavert G1 rootfigure
# ordinate 
iG1rootF_ord <- ordinate(physeq = INRAE_gen1_rootF, method="PCoA", distance= iG1rootF_BCdist)

# plot ordination
iG1rootF_PCoA <- plot_ordination(INRAE_gen1_rootF, iG1rootF_ord, color="Treatment") + theme(aspect.ratio=1)
iG1rootF_PCoA
# check figure and add axis values in code below

#change colors, title, etc.
iG1rootF_PCoA <- iG1rootF_PCoA  +   
  geom_point(aes(color=Treatment, shape=Treatment), size=3, alpha=1) +
  labs(title = "(FR) G1 Roots", x="PCoA1 [61.8%]", y="PCoA2 [17.7%]") +
  scale_color_manual(name = "G1 Treatment",
    values = c("#3399FF","#FFCC00"), labels = c("Control", "Drought"), 
    guide = guide_legend(override.aes = list(size = 3), alpha = 1) ) +
   scale_shape_manual(name = "G1 Treatment",
    values = c("circle", "triangle"), labels = c("Control", "Drought"), 
    guide = guide_legend(override.aes = list(size = 3), alpha = 1)) + my_theme + facet_wrap(~Genotype) +
  theme(legend.position="none", plot.tag = element_text(face="bold"))
iG1rootF_PCoA


# RedHawk G1 root figure
# ordinate 
iG1rootRH_ord <- ordinate(physeq = INRAE_gen1_rootRH, method="PCoA", distance= iG1rootRH_BCdist)

# plot ordination
iG1rootRH_PCoA <- plot_ordination(INRAE_gen1_rootRH, iG1rootRH_ord, color="Treatment") + theme(aspect.ratio=1)
iG1rootRH_PCoA
# check figure and add axis values in code below

#change colors, title, etc.
iG1rootRH_PCoA <- iG1rootRH_PCoA  +   
  geom_point(aes(color=Treatment, shape=Treatment), size=3, alpha=1) +
  labs(title = "(FR) G1 Roots", x="PCoA1 [40.9%]", y="PCoA2 [19.9%]") +
  scale_color_manual(name = "G1 Treatment",
    values = c("#3399FF","#FFCC00"), labels = c("Control", "Drought"), 
    guide = guide_legend(override.aes = list(size = 3), alpha = 1) ) +
   scale_shape_manual(name = "G1 Treatment",
    values = c("circle", "triangle"), labels = c("Control", "Drought"), 
    guide = guide_legend(override.aes = list(size = 3), alpha = 1)) + my_theme + facet_wrap(~Genotype) +
  theme(legend.position="none", plot.tag = element_text(face="bold"))
iG1rootRH_PCoA


# Generation 1 rhizosphere
perm_iG1rhizo <- how(nperm = 9999) 
meta_iG1rhizo = data.frame(sample_data(INRAE_gen1_rhizo))

iG1rhizo_BCdist = vegdist(otu_table(INRAE_gen1_rhizo), method = "bray", binary = FALSE)

set.seed(8)
iG1rhizo_BC_adonis <- adonis2(formula = iG1rhizo_BCdist ~ Genotype*G1_treatment, permutations = perm_iG1rhizo, data = meta_iG1rhizo)

iG1rhizo_BC_adonis
# G1_treatment  1  0.20079 0.08549 1.6993 0.0337 *

# Flavert G1 rhizo
perm_iG1rhizoF <- how(nperm = 9999) 
meta_iG1rhizoF = data.frame(sample_data(INRAE_gen1_rhizoF))

iG1rhizoF_BCdist = vegdist(otu_table(INRAE_gen1_rhizoF), method = "bray", binary = FALSE)

set.seed(8)
iG1rhizoF_BC_adonis <- adonis2(formula = iG1rhizoF_BCdist ~ G1_treatment, permutations = perm_iG1rhizoF, data = meta_iG1rhizoF)

iG1rhizoF_BC_adonis
# G1 treatment *

# RedHawk G1 rhizo
perm_iG1rhizoRH <- how(nperm = 9999) 
meta_iG1rhizoRH = data.frame(sample_data(INRAE_gen1_rhizoRH))

iG1rhizoRH_BCdist = vegdist(otu_table(INRAE_gen1_rhizoRH), method = "bray", binary = FALSE)

set.seed(8)
iG1rhizoRH_BC_adonis <- adonis2(formula = iG1rhizoRH_BCdist ~ G1_treatment, permutations = perm_iG1rhizoRH, data = meta_iG1rhizoRH)

iG1rhizoRH_BC_adonis
# NS


# Flavert G1 rhizo figure
# ordinate 
iG1rhizoF_ord <- ordinate(physeq = INRAE_gen1_rhizoF, method="PCoA", distance= iG1rhizoF_BCdist)

# plot ordination
iG1rhizoF_PCoA <- plot_ordination(INRAE_gen1_rhizoF, iG1rhizoF_ord, color="Treatment") + theme(aspect.ratio=1)
iG1rhizoF_PCoA
# check figure and add axis values in code below

#change colors, title, etc.
iG1rhizoF_PCoA <- iG1rhizoF_PCoA  +   
  geom_point(aes(color=Treatment, shape=Treatment), size=3, alpha=1) +
  labs(title = "(FR) G1 Rhizosphere", x="PCoA1 [32.1%]", y="PCoA2 [12%]", caption="Treatment *") +
  scale_color_manual(name = "G1 Treatment",
    values = c("#3399FF","#FFCC00"), labels = c("Control", "Drought"), 
    guide = guide_legend(override.aes = list(size = 3), alpha = 1) ) +
   scale_shape_manual(name = "G1 Treatment",
    values = c("circle", "triangle"), labels = c("Control", "Drought"), 
    guide = guide_legend(override.aes = list(size = 3), alpha = 1)) + my_theme + facet_wrap(~Genotype) + 
  theme(legend.position="none", plot.tag = element_text(face="bold"))
iG1rhizoF_PCoA


# RedHawk G1 rhizo figure
# ordinate 
iG1rhizoRH_ord <- ordinate(physeq = INRAE_gen1_rhizoRH, method="PCoA", distance= iG1rhizoRH_BCdist)

# plot ordination
iG1rhizoRH_PCoA <- plot_ordination(INRAE_gen1_rhizoRH, iG1rhizoRH_ord, color="Treatment") + theme(aspect.ratio=1)
iG1rhizoRH_PCoA
# check figure and add axis values in code below

#change colors, title, etc.
iG1rhizoRH_PCoA <- iG1rhizoRH_PCoA  +   
  geom_point(aes(color=Treatment, shape=Treatment), size=3, alpha=1) +
  labs(title = "(FR) G1 Rhizosphere", x="PCoA1 [33.4%]", y="PCoA2 [13.4%]") +
  scale_color_manual(name = "G1 Treatment",
    values = c("#3399FF","#FFCC00"), labels = c("Control", "Drought"), 
    guide = guide_legend(override.aes = list(size = 3), alpha = 1) ) +
   scale_shape_manual(name = "G1 Treatment",
    values = c("circle", "triangle"), labels = c("Control", "Drought"), 
    guide = guide_legend(override.aes = list(size = 3), alpha = 1)) + my_theme + facet_wrap(~Genotype)  + 
  theme(legend.position="none", plot.tag = element_text(face="bold"))
iG1rhizoRH_PCoA


######## Generation 2 roots

perm_iG2root <- how(nperm = 9999) 
meta_iG2root = data.frame(sample_data(INRAE_gen2_root))

iG2root_BCdist = vegdist(otu_table(INRAE_gen2_root), method = "bray", binary = FALSE)

set.seed(8)
iG2root_BC_adonis <- adonis2(formula = iG2root_BCdist ~ Genotype*G1_treatment*G2_treatment, permutations = perm_iG2root, data = meta_iG2root)

iG2root_BC_adonis
# Genotype ***
# G1:G2 *

# Flavert G2 root
perm_iG2rootF <- how(nperm = 9999) 
meta_iG2rootF = data.frame(sample_data(INRAE_gen2_rootF))

iG2rootF_BCdist = vegdist(otu_table(INRAE_gen2_rootF), method = "bray", binary = FALSE)

set.seed(8)
iG2rootF_BC_adonis <- adonis2(formula = iG2rootF_BCdist ~ G1_treatment*G2_treatment, permutations = perm_iG2rootF, data = meta_iG2rootF)

iG2rootF_BC_adonis
# G1:G2 **  (posthoc)

iG2rootF_pairwise <- pairwise.adonis2(iG2rootF_BCdist ~ G1_G2, data = meta_iG2rootF, permutations = perm_iG2rootF)
iG2rootF_pairwise
# Control_Drought_vs_Drought_Drought G1_G2     1  0.40809 0.42768 5.2309 0.0081 **
# Drought_Control_vs_Drought_Drought G1_G2     1  0.37378 0.33426 4.0168 0.0615


# RedHawk G2 root
perm_iG2rootRH <- how(nperm = 9999) 
meta_iG2rootRH = data.frame(sample_data(INRAE_gen2_rootRH))

iG2rootRH_BCdist = vegdist(otu_table(INRAE_gen2_rootRH), method = "bray", binary = FALSE)

set.seed(8)
iG2rootRH_BC_adonis <- adonis2(formula = iG2rootRH_BCdist ~ G1_treatment*G2_treatment, permutations = perm_iG2rootRH, data = meta_iG2rootRH)

iG2rootRH_BC_adonis
# NS


# Flavert G2 root figure
# ordinate 
iG2rootF_ord <- ordinate(physeq = INRAE_gen2_rootF, method="PCoA", distance= iG2rootF_BCdist)

# plot ordination
iG2rootF_PCoA <- plot_ordination(INRAE_gen2_rootF, iG2rootF_ord, color="G1_G2") + theme(aspect.ratio=1)
iG2rootF_PCoA
# check figure and add axis values in code below

#change colors, title, etc.
iG2rootF_PCoA <- iG2rootF_PCoA  +   
  geom_point(aes(color=G1_G2, shape=G1_treatment), size=3, alpha=1) +
  labs(title = "(FR) G2 Roots", x="PCoA1 [81.9%]", y="PCoA2 [10.8%]",
       caption = "G1 x G2 *")+
  scale_color_manual(name = "G1_G2 Consecutive\nTreatment",
    values = c("#3399FF","#99CCFF", "#FFCC00", "#FF6600"), labels = c("Control_Control", "Control_Drought", "Drought_Control", "Drought_Drought"), guide = guide_legend(override.aes = list(size = 3), alpha = 1)) +
   scale_shape_manual(name = "G1 Treatment",
    values = c("circle", "triangle"), labels = c("Control", "Drought"), guide = guide_legend(override.aes = list(size = 3), alpha = 1)) + my_theme + theme(legend.position="none", plot.tag = element_text(face="bold")) + facet_wrap(~Genotype)
iG2rootF_PCoA


# RedHawk G2 root figure
# ordinate 
iG2rootRH_ord <- ordinate(physeq = INRAE_gen2_rootRH, method="PCoA", distance= iG2rootRH_BCdist)

# plot ordination
iG2rootRH_PCoA <- plot_ordination(INRAE_gen2_rootRH, iG2rootRH_ord, color="G1_G2") + theme(aspect.ratio=1)
iG2rootRH_PCoA
# check figure and add axis values in code below

#change colors, title, etc.
iG2rootRH_PCoA <- iG2rootRH_PCoA  +   
  geom_point(aes(color=G1_G2, shape=G1_treatment), size=3, alpha=1) +
  labs(title = "(FR) G2 Roots", x="PCoA1 [86.4%]", y="PCoA2 [6.5%]") +
  scale_color_manual(name = "G1_G2 Consecutive\nTreatment",
    values = c("#3399FF","#99CCFF", "#FFCC00", "#FF6600"), labels = c("Control_Control", "Control_Drought", "Drought_Control", "Drought_Drought"), guide = guide_legend(override.aes = list(size = 3), alpha = 1)) +
   scale_shape_manual(name = "G1 Treatment",
    values = c("circle", "triangle"), labels = c("Control", "Drought"), guide = guide_legend(override.aes = list(size = 3), alpha = 1)) + my_theme + theme(legend.position="none", plot.tag = element_text(face="bold")) + facet_wrap(~Genotype)
iG2rootRH_PCoA




######## Generation 2 rhizo

perm_iG2rhizo <- how(nperm = 9999) 
meta_iG2rhizo = data.frame(sample_data(INRAE_gen2_rhizo))

iG2rhizo_BCdist = vegdist(otu_table(INRAE_gen2_rhizo), method = "bray", binary = FALSE)

set.seed(8)
iG2rhizo_BC_adonis <- adonis2(formula = iG2rhizo_BCdist ~ Genotype*G1_treatment*G2_treatment, permutations = perm_iG2rhizo, data = meta_iG2rhizo)

iG2rhizo_BC_adonis
#Genotype                            1   0.3823 0.06229 2.6091 0.0008 ***
#G1_treatment                        1   0.4253 0.06930 2.9027 0.0001 ***
#G2_treatment                        1   0.3301 0.05379 2.2530 0.0039 ** 
#Genotype:G1_treatment               1   0.4060 0.06616 2.7709 0.0005 ***

# Flavert G2 rhizo
perm_iG2rhizoF <- how(nperm = 9999) 
meta_iG2rhizoF = data.frame(sample_data(INRAE_gen2_rhizoF))

iG2rhizoF_BCdist = vegdist(otu_table(INRAE_gen2_rhizoF), method = "bray", binary = FALSE)

set.seed(8)
iG2rhizoF_BC_adonis <- adonis2(formula = iG2rhizoF_BCdist ~ G1_treatment*G2_treatment, permutations = perm_iG2rhizoF, data = meta_iG2rhizoF)

iG2rhizoF_BC_adonis
# G1 ***
# G2  1   0.2529 0.07526 1.6189 0.0514
# G1 x G2 0.2380 0.07083 1.5236 0.0555

# pairwise
iG2rhizoF_pairwise <- pairwise.adonis2(iG2rhizoF_BCdist ~ G1_G2, data = meta_iG2rhizoF, permutations = perm_iG2rhizoF)
iG2rhizoF_pairwise
# C_C vs C_D 0.23008 2.0919 0.0076 **
# C_C vs D_D 0.34351 4.1859 0.0088 **
# C_C vs D_C 0.39111 4.4963 0.0102 *
# C_D vs D_D 0.25361 2.3785 0.0083 **
# C_D vs D_C 0.27233 2.2455 0.0316 *

# RedHawk G2 rhizo
perm_iG2rhizoRH <- how(nperm = 9999) 
meta_iG2rhizoRH = data.frame(sample_data(INRAE_gen2_rhizoRH))

iG2rhizoRH_BCdist = vegdist(otu_table(INRAE_gen2_rhizoRH), method = "bray", binary = FALSE)

set.seed(8)
iG2rhizoRH_BC_adonis <- adonis2(formula = iG2rhizoRH_BCdist ~ G1_treatment*G2_treatment, permutations = perm_iG2rhizoRH, data = meta_iG2rhizoRH)

iG2rhizoRH_BC_adonis
# G2_treatment 1  0.20922 0.08737 1.5289 0.0624 .


# Flavert G2 rhizo figure
# ordinate 
iG2rhizoF_ord <- ordinate(physeq = INRAE_gen2_rhizoF, method="PCoA", distance= iG2rhizoF_BCdist)

# plot ordination
iG2rhizoF_PCoA <- plot_ordination(INRAE_gen2_rhizoF, iG2rhizoF_ord, color="G1_G2") + theme(aspect.ratio=1)
iG2rhizoF_PCoA
# check figure and add axis values in code below

#change colors, title, etc.
iG2rhizoF_PCoA <- iG2rhizoF_PCoA  +   
  geom_point(aes(color=G1_G2, shape=G1_treatment), size=3, alpha=1) +
  labs(title = "(FR) G2 Rhizosphere", x="PCoA1 [25.4%]", y="PCoA2 [19.9%]", 
       caption="G1 Treatment ***\nG2 Treatment [p-val=0.051]\nG1 x G2 [p-val=0.055]") +
  scale_color_manual(name = "G1_G2 Consecutive\nTreatment",
    values = c("#3399FF","#99CCFF", "#FFCC00", "#FF6600"), labels = c("Control_Control", "Control_Drought", "Drought_Control", "Drought_Drought"), guide = guide_legend(override.aes = list(size = 3), alpha = 1)) +
   scale_shape_manual(name = "G1 Treatment",
    values = c("circle", "triangle"), labels = c("Control", "Drought"), guide = guide_legend(override.aes = list(size = 3), alpha = 1)) + my_theme + theme(legend.position="none", plot.tag = element_text(face="bold"))  + facet_wrap(~Genotype)
iG2rhizoF_PCoA


# RedHawk G2 root figure
# ordinate 
iG2rhizoRH_ord <- ordinate(physeq = INRAE_gen2_rhizoRH, method="PCoA", distance= iG2rhizoRH_BCdist)

# plot ordination
iG2rhizoRH_PCoA <- plot_ordination(INRAE_gen2_rhizoRH, iG2rhizoRH_ord, color="G1_G2") + theme(aspect.ratio=1)
iG2rhizoRH_PCoA
# check figure and add axis values in code below

#change colors, title, etc.
iG2rhizoRH_PCoA <- iG2rhizoRH_PCoA  +   
  geom_point(aes(color=G1_G2, shape=G1_treatment), size=3, alpha=1) +
  labs(title = "(FR) G2 Rhizosphere", x="PCoA1 [21.7%]", y="PCoA2 [13.3%]")+
  scale_color_manual(name = "G1_G2 Consecutive\nTreatment",
    values = c("#3399FF","#99CCFF", "#FFCC00", "#FF6600"), labels = c("Control_Control", "Control_Drought", "Drought_Control", "Drought_Drought"), guide = guide_legend(override.aes = list(size = 3), alpha = 1)) +
   scale_shape_manual(name = "G1 Treatment",
    values = c("circle", "triangle"), labels = c("Control", "Drought"), guide = guide_legend(override.aes = list(size = 3), alpha = 1)) + my_theme + theme(plot.tag = element_text(face="bold"), legend.position="none") + facet_wrap(~Genotype)
iG2rhizoRH_PCoA

```


# Beta diversity MSU
```{r}
MSU_phyloseq <- readRDS(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/updated_MSU_CD_phyloseq.rds")

MSU_gen1 <- MSU_phyloseq %>%
  ps_filter(Generation == "G1")

MSU_gen1_root <- MSU_gen1 %>% 
  ps_filter(Compartment == "Root")

MSU_gen1_rhizo <- MSU_gen1 %>% 
  ps_filter(Compartment == "Rhizosphere")

MSU_gen2 <- MSU_phyloseq %>%
  ps_filter(Generation == "G2")

MSU_gen2_root <- MSU_gen2 %>% 
  ps_filter(Compartment == "Root")

MSU_gen2_rhizo <- MSU_gen2 %>% 
  ps_filter(Compartment == "Rhizosphere")

#root_gen1_BCdist_16S = vegdist(otu_table(root_16S_gen1), method = "bray", binary = FALSE, diag = TRUE, upper = TRUE)
# ordinate 
#root_g1_ord_16S <- ordinate(physeq = root_16S_gen1, method="PCoA", distance=root_gen1_BCdist_16S)
# plot ordination
#G1_roots_PCoA_16S <- plot_ordination(root_16S_gen1, root_g1_ord_16S, color="Treatment") + theme(aspect.ratio=1)
#G1_roots_PCoA_16S + facet_wrap(~Group, scales="free")
# Try another way
#root_g1_ord_16S_2 <- ordinate(physeq = root_16S_gen1, method="PCoA", distance="bray")
# plot ordination
#G1_roots_PCoA_16S_2 <- plot_ordination(root_16S_gen1, root_g1_ord_16S_2, color="Treatment") + theme(aspect.ratio=1)
#G1_roots_PCoA_16S_2 + facet_wrap(~Group, scales="free")


# Generation 1 roots
perm_mG1root <- how(nperm = 9999) 
meta_mG1root = data.frame(sample_data(MSU_gen1_root))
#setBlocks(perm_g1_root) <- with(root_gen1_metadata, growth_chamber)

mG1root_BCdist = vegdist(t(otu_table(MSU_gen1_root)), method = "bray", binary = FALSE)

set.seed(8)
mG1root_BC_adonis <- adonis2(formula = mG1root_BCdist ~ G1_treatment, permutations = perm_mG1root, data = meta_mG1root)

mG1root_BC_adonis
# NS

#figure

# ordinate 
mG1root_ord <- ordinate(physeq = MSU_gen1_root, method="PCoA", distance= mG1root_BCdist)

# plot ordination
mG1root_PCoA <- plot_ordination(MSU_gen1_root, mG1root_ord, color="Treatment") + theme(aspect.ratio=1)
# check figure and add axis values in code below

#change colors, title, etc.
mG1root_PCoA <- mG1root_PCoA  +   
  geom_point(aes(color=Treatment, shape=Treatment), size=3, alpha=1) +
  labs(title = "(USA) G1 Roots", x="PCoA1 [37.3%]", y="PCoA2 [22.4%]") +
  scale_color_manual(name = "G1 Treatment",
    values = c("#3399FF","#FFCC00"), labels = c("Control", "Drought"), 
    guide = guide_legend(override.aes = list(size = 3), alpha = 1) ) +
   scale_shape_manual(name = "G1 Treatment",
    values = c("circle", "triangle"), labels = c("Control", "Drought"), 
    guide = guide_legend(override.aes = list(size = 3), alpha = 1)) + my_theme + 
  theme(plot.tag = element_text(face="bold")) + facet_wrap(~Genotype)
mG1root_PCoA


# Generation 2 roots, block by planting group

perm_mG2root <- how(nperm = 9999) 
meta_mG2root = data.frame(sample_data(MSU_gen2_root))

str(meta_mG2root)
meta_mG2root$planting_group <- as.factor(meta_mG2root$planting_group)

setBlocks(perm_mG2root) <- with(meta_mG2root, planting_group)

#root_gen2_metadata <- root_gen2_metadata %>% rownames_to_column(var="SampleID")

mG2root_BCdist = vegdist(t(otu_table(MSU_gen2_root)), method = "bray", binary = FALSE)

set.seed(8)
mG2root_BC_adonis <- adonis2(formula = mG2root_BCdist ~ G2_treatment*G1_treatment, permutations = perm_mG2root, data = meta_mG2root)

mG2root_BC_adonis

# PERMANOVA G2 treatment *
# G1 treatment NS

#make figure
mG2root_cap_ord <- ordinate(physeq = MSU_gen2_root, method="CAP", distance= mG2root_BCdist, formula= ~ 1 + Condition(planting_group))

mG2root_cap.ord.df = data.frame(vegan::scores(mG2root_cap_ord, display="sites")) %>%
  tibble::rownames_to_column(var="SampleID") %>%
  select(SampleID, MDS1, MDS2) %>%
  left_join(meta_mG2root, by="SampleID") 

#cap.ord.df %>%
#  filter((planting_group == 2 & MDS1 < 0) | MDS2 < -1.5)

mG2root_eigvec = vegan::eigenvals(mG2root_cap_ord)
mG2root_fracvar = round(mG2root_eigvec/sum(mG2root_eigvec)*100, 2)



mG2root_cap_plot = ggplot(data=mG2root_cap.ord.df, aes(x=MDS1, y=MDS2)) +
  geom_point(aes(color=G1_G2, shape=G1_treatment), size=3, alpha=1) +
  labs(x=paste("PCoA1 [", mG2root_fracvar[1], "%]", sep=""),
       y=paste("PCoA2 [", mG2root_fracvar[2], "%]", sep=""), caption="G2 Treatment *") +
  scale_color_manual(name = "G1_G2 Consecutive\nTreatment",
    values = c("#3399FF","#99CCFF", "#FFCC00", "#FF6600"), labels = c("Control_Control", "Control_Drought", "Drought_Control", "Drought_Drought"), guide = guide_legend(override.aes = list(size = 3), alpha = 1)) +
   scale_shape_manual(name = "G1 Treatment",
    values = c("circle", "triangle"), labels = c("Control", "Drought"), guide = guide_legend(override.aes = list(size = 3), alpha = 1)) +
  # add a title and delete the automatic caption
  labs(title = "(USA) G2 Roots") + my_theme + theme(aspect.ratio=1, plot.tag = element_text(face="bold")) + facet_wrap(~Genotype)
  
mG2root_cap_plot


# Generation 1 rhizosphere
perm_mG1rhizo <- how(nperm = 9999) 
meta_mG1rhizo = data.frame(sample_data(MSU_gen1_rhizo))
#setBlocks(perm_g1_root) <- with(root_gen1_metadata, growth_chamber)

mG1rhizo_BCdist = vegdist(t(otu_table(MSU_gen1_rhizo)), method = "bray", binary = FALSE)

set.seed(8)
mG1rhizo_BC_adonis <- adonis2(formula = mG1rhizo_BCdist ~ G1_treatment, permutations = perm_mG1rhizo, data = meta_mG1rhizo)

mG1rhizo_BC_adonis
# treatment NS


#figure

# ordinate 
mG1rhizo_ord <- ordinate(physeq = MSU_gen1_rhizo, method="PCoA", distance= mG1rhizo_BCdist)

# plot ordination
mG1rhizo_PCoA <- plot_ordination(MSU_gen1_rhizo, mG1rhizo_ord, color="Treatment") + theme(aspect.ratio=1)
# check figure and add axis values in code below

#change colors, title, etc.
mG1rhizo_PCoA <- mG1rhizo_PCoA  +   
  geom_point(aes(color=Treatment, shape=Treatment), size=3, alpha=1) +
  labs(title = "(USA) G1 Rhizosphere", x="PCoA1 [30.9%]", y="PCoA2 [12.2%]") +
  scale_color_manual(name = "G1 Treatment",
    values = c("#3399FF","#FFCC00"), labels = c("Control", "Drought"), 
    guide = guide_legend(override.aes = list(size = 3), alpha = 1) ) +
   scale_shape_manual(name = "G1 Treatment",
    values = c("circle", "triangle"), labels = c("Control", "Drought"), 
    guide = guide_legend(override.aes = list(size = 3), alpha = 1)) + my_theme + 
  theme(legend.position="none", plot.tag = element_text(face="bold")) + facet_wrap(~Genotype)
mG1rhizo_PCoA


# Generation 2 rhizo, block by planting group

perm_mG2rhizo <- how(nperm = 9999) 
meta_mG2rhizo = data.frame(sample_data(MSU_gen2_rhizo))

str(meta_mG2rhizo)
meta_mG2rhizo$planting_group <- as.factor(meta_mG2rhizo$planting_group)

setBlocks(perm_mG2rhizo) <- with(meta_mG2rhizo, planting_group)

#root_gen2_metadata <- root_gen2_metadata %>% rownames_to_column(var="SampleID")

mG2rhizo_BCdist = vegdist(t(otu_table(MSU_gen2_rhizo)), method = "bray", binary = FALSE)

set.seed(8)
mG2rhizo_BC_adonis <- adonis2(formula = mG2rhizo_BCdist ~ G2_treatment*G1_treatment, permutations = perm_mG2rhizo, data = meta_mG2rhizo)

mG2rhizo_BC_adonis

# G2 treatment NS
# G1 treatment NS

#make figure
mG2rhizo_cap_ord <- ordinate(physeq = MSU_gen2_rhizo, method="CAP", distance= mG2rhizo_BCdist, formula= ~ 1 + Condition(planting_group))

mG2rhizo_cap.ord.df = data.frame(vegan::scores(mG2rhizo_cap_ord, display="sites")) %>%
  tibble::rownames_to_column(var="SampleID") %>%
  select(SampleID, MDS1, MDS2) %>%
  left_join(meta_mG2rhizo, by="SampleID") 

#cap.ord.df %>%
#  filter((planting_group == 2 & MDS1 < 0) | MDS2 < -1.5)

mG2rhizo_eigvec = vegan::eigenvals(mG2rhizo_cap_ord)
mG2rhizo_fracvar = round(mG2rhizo_eigvec/sum(mG2rhizo_eigvec)*100, 2)



mG2rhizo_cap_plot = ggplot(data=mG2rhizo_cap.ord.df, aes(x=MDS1, y=MDS2)) +
  geom_point(aes(color=G1_G2, shape=G1_treatment), size=3, alpha=1) +
  labs(x=paste("PCoA1 [", mG2rhizo_fracvar[1], "%]", sep=""),
       y=paste("PCoA2 [", mG2rhizo_fracvar[2], "%]", sep="")) +
  scale_color_manual(name = "G1_G2 Consecutive\nTreatment",
    values = c("#3399FF","#99CCFF", "#FFCC00", "#FF6600"), labels = c("Control_Control", "Control_Drought", "Drought_Control", "Drought_Drought"), guide = guide_legend(override.aes = list(size = 3), alpha = 1)) +
   scale_shape_manual(name = "G1 Treatment",
    values = c("circle", "triangle"), labels = c("Control", "Drought"), guide = guide_legend(override.aes = list(size = 3), alpha = 1)) +
  # add a title and delete the automatic caption
  labs(title = "(USA) G2 Rhizosphere") + my_theme + theme(aspect.ratio=1, legend.position="none", plot.tag = element_text(face="bold")) + facet_wrap(~Genotype)
  
mG2rhizo_cap_plot
```

# Compiled beta diversity and dispersion figures
```{r}
gen1_PCoA_panels <- (iG1rootF_PCoA + iG1rootRH_PCoA + mG1root_PCoA) / (iG1rhizoF_PCoA + iG1rhizoRH_PCoA + mG1rhizo_PCoA) + plot_annotation(tag_levels = 'A')

gen1_PCoA_panels

ggsave(gen1_PCoA_panels, file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/gen1_PCoA.png", width = 9, height = 6, units="in")

## Fig.6
ggsave(plot=gen1_PCoA_panels, "/PATH/Fig.6.tif",
       width=9, height=6,
       device = "tiff",
       units= "in", dpi = 400,
       compression="lzw", bg= "white")



gen2_PCoA_panels <- (iG2rootF_PCoA + iG2rootRH_PCoA + mG2root_cap_plot) / (iG2rhizoF_PCoA + iG2rhizoRH_PCoA + mG2rhizo_cap_plot) + plot_annotation(tag_levels = 'A')

gen2_PCoA_panels

ggsave(gen2_PCoA_panels, file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/gen2_PCoA.png", width = 12, height = 8, units="in")

## Fig.7
ggsave(plot=gen2_PCoA_panels, "/PATH/Fig.7.tif",
       width=12, height=8,
       device = "tiff",
       units= "in", dpi = 400,
       compression="lzw", bg= "white")

beta_dispersion_plots <- iG2rootRH_disper_plot+iG2rhizoF_disper_plot + mG2rhizo_disper_plot2 + plot_annotation(tag_levels = 'A')

beta_dispersion_plots

beta_dispersion <- iG2rootRH_disper_plot+iG2rhizoF_disper_plot + mG2rhizo_disper_plot2 + plot_annotation(tag_levels = 'A')

ggsave(beta_dispersion, file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/beta_dispersion_Fig7.png", width = 10, height = 5, units="in")

## Fig. 8
ggsave(plot=beta_dispersion_plots, "/PATH/Fig.8.tif",
       width=10, height=5,
       device = "tiff",
       units= "in", dpi = 400,
       compression="lzw", bg= "white")

```

# Beta dispersion
```{r}
#run betadisper on BC distribution by G1 treatment using geometric median method

# MSU G1 root
mG1root_betadisper <- betadisper(mG1root_BCdist, group=as.factor(meta_mG1root$G1_treatment), type = "median")
mG1root_betadisper

permutest(mG1root_betadisper, pairwise=TRUE, permutations=999)
TukeyHSD(mG1root_betadisper)
#NS

boxplot(mG1root_betadisper)

# get betadisper data ####
mG1root_BDdata <- get_betadisper_data(mG1root_betadisper)

# do some transformations on the data
mG1root_BDdata$eigenvalue <- mutate(mG1root_BDdata$eigenvalue, percent = eig/sum(eig))

# add convex hull points ####
# this could be put in a function
mG1root_BDdata$chull <- group_by(mG1root_BDdata$eigenvector, group) %>%
  do(data.frame(PCoA1 = .$PCoA1[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])],
                PCoA2 = .$PCoA2[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])])) %>%
  data.frame()

# combine centroid and eigenvector dataframes for plotting (only need this if we want to plot the centroids)
mG1root_BDdata_lines <- merge(select(mG1root_BDdata$centroids, group, PCoA1, PCoA2), select(mG1root_BDdata$eigenvector, group, PCoA1, PCoA2), by = c('group'))

# Now the dataframes are all ready to be completely customisable in ggplot
mG1root_disper_plot <- ggplot(mG1root_BDdata$distances, aes(group, distances, col = group)) +
  geom_boxplot(outlier.shape = NA, position = position_dodge(width = 0.55)) +
  geom_point(size=3, shape=21, fill="white") +
  my_theme +
  scale_color_manual(name="G1 Treatment", values = c("#3399FF","#FFCC00"), labels = c("Control", 'Drought')) +
  ylab('Distance to Median') + xlab("G1 Treatment") +
  scale_x_discrete(labels = c("Control", 'Drought')) +
  theme(legend.position = "none")
mG1root_disper_plot


# MSU G1 rhizo
mG1rhizo_betadisper <- betadisper(mG1rhizo_BCdist, group=as.factor(meta_mG1rhizo$G1_treatment), type = "median")
mG1rhizo_betadisper

permutest(mG1rhizo_betadisper, pairwise=TRUE, permutations=999)
TukeyHSD(mG1rhizo_betadisper)
#NS

boxplot(mG1rhizo_betadisper)

# get betadisper data ####
mG1rhizo_BDdata <- get_betadisper_data(mG1rhizo_betadisper)

mG1rhizo_BDdata$eigenvalue <- mutate(mG1rhizo_BDdata$eigenvalue, percent = eig/sum(eig))

mG1rhizo_BDdata$chull <- group_by(mG1rhizo_BDdata$eigenvector, group) %>%
  do(data.frame(PCoA1 = .$PCoA1[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])],
                PCoA2 = .$PCoA2[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])])) %>%
  data.frame()

# Now the dataframes are all ready to be completely customisable in ggplot
mG1rhizo_disper_plot <- ggplot(mG1rhizo_BDdata$distances, aes(group, distances, col = group)) +
  geom_boxplot(outlier.shape = NA, position = position_dodge(width = 0.55)) +
  geom_point(size=3, shape=21, fill="white") +
  my_theme +
  scale_color_manual(name="G1 Treatment", values = c("#3399FF","#FFCC00"), labels = c("Control", 'Drought')) +
  ylab('Distance to Median') + xlab("G1 Treatment") +
  scale_x_discrete(labels = c("Control", 'Drought')) +
  theme(legend.position = "none")
mG1rhizo_disper_plot


# INRAE G1 root

# Flavert
iG1rootF_betadisper <- betadisper(iG1rootF_BCdist, group=as.factor(meta_iG1rootF$G1_treatment), type = "median")
iG1rootF_betadisper

permutest(iG1rootF_betadisper, pairwise=TRUE, permutations=999)
TukeyHSD(iG1rootF_betadisper)
#NS

boxplot(iG1rootF_betadisper)

# get betadisper data ####
iG1rootF_BDdata <- get_betadisper_data(iG1rootF_betadisper)

iG1rootF_BDdata$eigenvalue <- mutate(iG1rootF_BDdata$eigenvalue, percent = eig/sum(eig))

iG1rootF_BDdata$chull <- group_by(iG1rootF_BDdata$eigenvector, group) %>%
  do(data.frame(PCoA1 = .$PCoA1[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])],
                PCoA2 = .$PCoA2[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])])) %>%
  data.frame()

# Now the dataframes are all ready to be completely customisable in ggplot
iG1rootF_disper_plot <- ggplot(iG1rootF_BDdata$distances, aes(group, distances, col = group)) +
  geom_boxplot(outlier.shape = NA, position = position_dodge(width = 0.55)) +
  geom_point(size=3, shape=21, fill="white") +
  my_theme +
  scale_color_manual(name="G1 Treatment", values = c("#3399FF","#FFCC00"), labels = c("Control", 'Drought')) +
  ylab('Distance to Median') + xlab("G1 Treatment") +
  scale_x_discrete(labels = c("Control", 'Drought')) +
  theme(legend.position = "none")
iG1rootF_disper_plot

# RedHawk
iG1rootRH_betadisper <- betadisper(iG1rootRH_BCdist, group=as.factor(meta_iG1rootRH$G1_treatment), type = "median")
iG1rootRH_betadisper

permutest(iG1rootRH_betadisper, pairwise=TRUE, permutations=999)
TukeyHSD(iG1rootRH_betadisper)
#NS

boxplot(iG1rootRH_betadisper)

# get betadisper data ####
iG1rootRH_BDdata <- get_betadisper_data(iG1rootRH_betadisper)

iG1rootRH_BDdata$eigenvalue <- mutate(iG1rootRH_BDdata$eigenvalue, percent = eig/sum(eig))

iG1rootRH_BDdata$chull <- group_by(iG1rootRH_BDdata$eigenvector, group) %>%
  do(data.frame(PCoA1 = .$PCoA1[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])],
                PCoA2 = .$PCoA2[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])])) %>%
  data.frame()

# Now the dataframes are all ready to be completely customisable in ggplot
iG1rootRH_disper_plot <- ggplot(iG1rootRH_BDdata$distances, aes(group, distances, col = group)) +
  geom_boxplot(outlier.shape = NA, position = position_dodge(width = 0.55)) +
  geom_point(size=3, shape=21, fill="white") +
  my_theme +
  scale_color_manual(name="G1 Treatment", values = c("#3399FF","#FFCC00"), labels = c("Control", 'Drought')) +
  ylab('Distance to Median') + xlab("G1 Treatment") +
  scale_x_discrete(labels = c("Control", 'Drought')) +
  theme(legend.position = "none")
iG1rootRH_disper_plot



# INRAE G1 rhizo

# Flavert
iG1rhizoF_betadisper <- betadisper(iG1rhizoF_BCdist, group=as.factor(meta_iG1rhizoF$G1_treatment), type = "median")
iG1rhizoF_betadisper

permutest(iG1rhizoF_betadisper, pairwise=TRUE, permutations=999)
TukeyHSD(iG1rhizoF_betadisper)
#NS

boxplot(iG1rhizoF_betadisper)

# get betadisper data ####
iG1rhizoF_BDdata <- get_betadisper_data(iG1rhizoF_betadisper)

iG1rhizoF_BDdata$eigenvalue <- mutate(iG1rhizoF_BDdata$eigenvalue, percent = eig/sum(eig))

iG1rhizoF_BDdata$chull <- group_by(iG1rhizoF_BDdata$eigenvector, group) %>%
  do(data.frame(PCoA1 = .$PCoA1[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])],
                PCoA2 = .$PCoA2[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])])) %>%
  data.frame()

# Now the dataframes are all ready to be completely customisable in ggplot
iG1rhizoF_disper_plot <- ggplot(iG1rhizoF_BDdata$distances, aes(group, distances, col = group)) +
  geom_boxplot(outlier.shape = NA, position = position_dodge(width = 0.55)) +
  geom_point(size=3, shape=21, fill="white") +
  my_theme +
  scale_color_manual(name="G1 Treatment", values = c("#3399FF","#FFCC00"), labels = c("Control", 'Drought')) +
  ylab('Distance to Median') + xlab("G1 Treatment") +
  scale_x_discrete(labels = c("Control", 'Drought')) +
  theme(legend.position = "none")
iG1rhizoF_disper_plot


# RedHawk
iG1rhizoRH_betadisper <- betadisper(iG1rhizoRH_BCdist, group=as.factor(meta_iG1rhizoRH$G1_treatment), type = "median")
iG1rhizoRH_betadisper

permutest(iG1rhizoRH_betadisper, pairwise=TRUE, permutations=999)
TukeyHSD(iG1rhizoRH_betadisper)
#NS

boxplot(iG1rhizoRH_betadisper)

# get betadisper data ####
iG1rhizoRH_BDdata <- get_betadisper_data(iG1rhizoRH_betadisper)

iG1rhizoRH_BDdata$eigenvalue <- mutate(iG1rhizoRH_BDdata$eigenvalue, percent = eig/sum(eig))

iG1rhizoRH_BDdata$chull <- group_by(iG1rhizoRH_BDdata$eigenvector, group) %>%
  do(data.frame(PCoA1 = .$PCoA1[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])],
                PCoA2 = .$PCoA2[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])])) %>%
  data.frame()

# Now the dataframes are all ready to be completely customisable in ggplot
iG1rhizoRH_disper_plot <- ggplot(iG1rhizoRH_BDdata$distances, aes(group, distances, col = group)) +
  geom_boxplot(outlier.shape = NA, position = position_dodge(width = 0.55)) +
  geom_point(size=3, shape=21, fill="white") +
  my_theme +
  scale_color_manual(name="G1 Treatment", values = c("#3399FF","#FFCC00"), labels = c("Control", 'Drought')) +
  ylab('Distance to Median') + xlab("G1 Treatment") +
  scale_x_discrete(labels = c("Control", 'Drought')) +
  theme(legend.position = "none")
iG1rhizoRH_disper_plot



# Generation 2, test G1 treatment, G2 treatment, and G1_G2 groups

# MSU G2 root
mG2root_betadisper_G1 <- betadisper(mG2root_BCdist, group=as.factor(meta_mG2root$G1_treatment), type = "median")

mG2root_betadisper_G2 <- betadisper(mG2root_BCdist, group=as.factor(meta_mG2root$G2_treatment), type = "median")

mG2root_betadisper_G1G2 <- betadisper(mG2root_BCdist, group=as.factor(meta_mG2root$G1_G2), type = "median")

permutest(mG2root_betadisper_G1, pairwise=TRUE, permutations=999)
permutest(mG2root_betadisper_G2, pairwise=TRUE, permutations=999)
permutest(mG2root_betadisper_G1G2, pairwise=TRUE, permutations=999)
TukeyHSD(mG2root_betadisper_G1G2)
# Nothing is significant

boxplot(mG2root_betadisper_G1G2)

# get betadisper data ####
mG2root_BDdata <- get_betadisper_data(mG2root_betadisper_G1G2)

mG2root_BDdata$eigenvalue <- mutate(mG2root_BDdata$eigenvalue, percent = eig/sum(eig))

mG2root_BDdata$chull <- group_by(mG2root_BDdata$eigenvector, group) %>%
  do(data.frame(PCoA1 = .$PCoA1[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])],
                PCoA2 = .$PCoA2[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])])) %>%
  data.frame()

# Now the dataframes are all ready to be completely customisable in ggplot
mG2root_disper_plot <- ggplot(mG2root_BDdata$distances, aes(group, distances, col = group)) +
  geom_boxplot(outlier.shape = NA, position = position_dodge(width = 0.55)) +
  geom_point(size=3, shape=21, fill="white") +
  my_theme +
  scale_color_manual(name="G1_G2 Consecutive Treatment", values = c("#3399FF","#99CCFF", "#FFCC00", "#FF6600"), labels = c("Control_Control", "Control_Drought", "Drought_Control", "Drought_Drought")) +
  ylab('Distance to Median') + xlab("G1_G2 Consecutive Treatment") +
  scale_x_discrete(labels = c("C_C", 'C_D', "D_C", "D_D")) +
  theme(legend.position = "none")
mG2root_disper_plot


# MSU G2 rhizo
mG2rhizo_betadisper_G1 <- betadisper(mG2rhizo_BCdist, group=as.factor(meta_mG2rhizo$G1_treatment), type = "median")

mG2rhizo_betadisper_G2 <- betadisper(mG2rhizo_BCdist, group=as.factor(meta_mG2rhizo$G2_treatment), type = "median")

mG2rhizo_betadisper_G1G2 <- betadisper(mG2rhizo_BCdist, group=as.factor(meta_mG2rhizo$G1_G2), type = "median")

permutest(mG2rhizo_betadisper_G1, pairwise=TRUE, permutations=999)
permutest(mG2rhizo_betadisper_G2, pairwise=TRUE, permutations=999)
# G2 treatment *
permutest(mG2rhizo_betadisper_G1G2, pairwise=TRUE, permutations=999)
TukeyHSD(mG2rhizo_betadisper_G1G2)
# C_D and D_C are significant * with permutest,but NS with Tukey?

boxplot(mG2rhizo_betadisper_G1G2)

# get betadisper data ####
mG2rhizo_BDdata <- get_betadisper_data(mG2rhizo_betadisper_G1G2)

mG2rhizo_BDdata$eigenvalue <- mutate(mG2rhizo_BDdata$eigenvalue, percent = eig/sum(eig))

mG2rhizo_BDdata$chull <- group_by(mG2rhizo_BDdata$eigenvector, group) %>%
  do(data.frame(PCoA1 = .$PCoA1[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])],
                PCoA2 = .$PCoA2[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])])) %>%
  data.frame()

# Now the dataframes are all ready to be completely customisable in ggplot
mG2rhizo_BD_anno <- data.frame(x1 = c(2), x2 = c(3), 
                   y1 = c(0.48), y2 = c(0.485), 
                   xstar = c(2.5), ystar = c(0.487),
                   lab = c("*"))

mG2rhizo_disper_plot <- ggplot(mG2rhizo_BDdata$distances, aes(group, distances, col = group)) +
  geom_boxplot(outlier.shape = NA, position = position_dodge(width = 0.55)) +
  geom_point(size=3, shape=21, fill="white") +
  my_theme + labs(title="(USA) G2 RedHawk Rhizosphere") +
  scale_color_manual(name="G1_G2 Consecutive\nTreatment", values = c("#3399FF","#99CCFF", "#FFCC00", "#FF6600"), labels = c("Control_Control", "Control_Drought", "Drought_Control", "Drought_Drought")) +
  ylab('') + xlab("") +
  scale_x_discrete(labels = c("C_C", 'C_D', "D_C", "D_D")) + 
  labs(caption="G2 Treatment *") + 
  theme(plot.tag = element_text(face="bold"),
        plot.caption = element_text(size=13, face="italic")) 
mG2rhizo_disper_plot2 <- mG2rhizo_disper_plot + 
  geom_text(inherit.aes=FALSE, data = mG2rhizo_BD_anno, aes(x = xstar,  y = ystar, label = lab), size=6) +
  geom_segment(inherit.aes=FALSE, data = mG2rhizo_BD_anno, aes(x = x1, xend = x1, 
           y = y1, yend = y2),
           colour = "black") +
  geom_segment(inherit.aes=FALSE, data = mG2rhizo_BD_anno, aes(x = x2, xend = x2, 
           y = y1, yend = y2),
           colour = "black") +
  geom_segment(inherit.aes=FALSE, data = mG2rhizo_BD_anno, aes(x = x1, xend = x2, 
           y = y2, yend = y2),
           colour = "black")
mG2rhizo_disper_plot2

### Inrae
# G2 root - Flavert
iG2rootF_betadisper_G1 <- betadisper(iG2rootF_BCdist, group=as.factor(meta_iG2rootF$G1_treatment), type = "median")

iG2rootF_betadisper_G2 <- betadisper(iG2rootF_BCdist, group=as.factor(meta_iG2rootF$G2_treatment), type = "median")

iG2rootF_betadisper_G1G2 <- betadisper(iG2rootF_BCdist, group=as.factor(meta_iG2rootF$G1_G2), type = "median")
set.seed(8)
permutest(iG2rootF_betadisper_G1, pairwise=TRUE, permutations=999)
set.seed(8)
permutest(iG2rootF_betadisper_G2, pairwise=TRUE, permutations=999)
set.seed(8)
permutest(iG2rootF_betadisper_G1G2, pairwise=TRUE, permutations=999)
TukeyHSD(iG2rootF_betadisper_G1G2)
# NS

boxplot(iG2rootF_betadisper_G1G2)

# get betadisper data ####
iG2rootF_BDdata <- get_betadisper_data(iG2rootF_betadisper_G1G2)

iG2rootF_BDdata$eigenvalue <- mutate(iG2rootF_BDdata$eigenvalue, percent = eig/sum(eig))

iG2rootF_BDdata$chull <- group_by(iG2rootF_BDdata$eigenvector, group) %>%
  do(data.frame(PCoA1 = .$PCoA1[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])],
                PCoA2 = .$PCoA2[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])])) %>%
  data.frame()

# Now the dataframes are all ready to be completely customisable in ggplot

#iG2rootF_BD_anno <- data.frame(x1 = c(1), x2 = c(2), 
                  # y1 = c(0.49), y2 = c(0.51), 
                  # xstar = c(1.5), ystar = c(0.52),
                  # lab = c("*"))

iG2rootF_disper_plot <- ggplot(iG2rootF_BDdata$distances, aes(group, distances, col = group)) +
  geom_boxplot(outlier.shape = NA, position = position_dodge(width = 0.55)) +
  geom_point(size=3, shape=21, fill="white") +
  my_theme + labs(title="(FR) G2 Flavert Roots") +
  scale_color_manual(name="G1_G2 Consecutive Treatment", values = c("#3399FF","#99CCFF", "#FFCC00", "#FF6600"), labels = c("Control_Control", "Control_Drought", "Drought_Control", "Drought_Drought")) +
  ylab('Distance to Median') + xlab("") +
  scale_x_discrete(labels = c("C_C", 'C_D', "D_C", "D_D")) +
  theme(legend.position = "none", plot.tag = element_text(face="bold")) 

  #geom_text(inherit.aes=FALSE, data = iG2rootF_BD_anno, aes(x = xstar,  y = ystar, label = lab), size=6) +
  #geom_segment(inherit.aes=FALSE, data = iG2rootF_BD_anno, aes(x = x1, xend = x1, 
           #y = y1, yend = y2),
           #colour = "black") +
  #geom_segment(inherit.aes=FALSE, data = iG2rootF_BD_anno, aes(x = x2, xend = x2, 
           #y = y1, yend = y2),
           #colour = "black") +
  #geom_segment(inherit.aes=FALSE, data = iG2rootF_BD_anno, aes(x = x1, xend = x2, 
           #y = y2, yend = y2),
           #colour = "black")
iG2rootF_disper_plot

# G2 root - RedHawk
iG2rootRH_betadisper_G1 <- betadisper(iG2rootRH_BCdist, group=as.factor(meta_iG2rootRH$G1_treatment), type = "median")

iG2rootRH_betadisper_G2 <- betadisper(iG2rootRH_BCdist, group=as.factor(meta_iG2rootRH$G2_treatment), type = "median")

iG2rootRH_betadisper_G1G2 <- betadisper(iG2rootRH_BCdist, group=as.factor(meta_iG2rootRH$G1_G2), type = "median")
set.seed(8)
permutest(iG2rootRH_betadisper_G1, pairwise=TRUE, permutations=999)
set.seed(8)
permutest(iG2rootRH_betadisper_G2, pairwise=TRUE, permutations=999)
set.seed(8)
permutest(iG2rootRH_betadisper_G1G2, pairwise=TRUE, permutations=999)
# Drought_Control vs Control_Drought *
TukeyHSD(iG2rootRH_betadisper_G1G2)
# But Tukey is NS?

boxplot(iG2rootRH_betadisper_G1G2)

# get betadisper data ####
iG2rootRH_BDdata <- get_betadisper_data(iG2rootRH_betadisper_G1G2)

iG2rootRH_BDdata$eigenvalue <- mutate(iG2rootRH_BDdata$eigenvalue, percent = eig/sum(eig))

iG2rootRH_BDdata$chull <- group_by(iG2rootRH_BDdata$eigenvector, group) %>%
  do(data.frame(PCoA1 = .$PCoA1[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])],
                PCoA2 = .$PCoA2[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])])) %>%
  data.frame()

# Now the dataframes are all ready to be completely customisable in ggplot

iG2rootRH_BD_anno <- data.frame(x1 = c(2), x2 = c(3), 
                               y1 = c(0.26), y2 = c(0.28),
                               xstar = c(2.5), ystar = c(0.3),
                               lab = c("*"))

iG2rootRH_disper_plot <- ggplot(iG2rootRH_BDdata$distances, aes(group, distances, col = group)) +
  geom_boxplot(outlier.shape = NA, position = position_dodge(width = 0.55)) +
  geom_point(size=3, shape=21, fill="white") +
  my_theme +
  scale_color_manual(name="G1_G2 Consecutive Treatment", values = c("#3399FF","#99CCFF", "#FFCC00", "#FF6600"), labels = c("Control_Control", "Control_Drought", "Drought_Control", "Drought_Drought")) +
  ylab('Distance to Median') + xlab("G1_G2 Consecutive Treatment") +
  scale_x_discrete(labels = c("C_C", 'C_D', "D_C", "D_D")) +
  theme(legend.position = "none", axis.title.x = element_blank(), plot.tag = element_text(face="bold")) +
  geom_text(inherit.aes=FALSE, data = iG2rootRH_BD_anno, aes(x = xstar,  y = ystar, label = lab), size=6) +
  geom_segment(inherit.aes=FALSE, data = iG2rootRH_BD_anno, aes(x = x1, xend = x1, y = y1, yend = y2), 
               colour = "black") +
  geom_segment(inherit.aes=FALSE, data = iG2rootRH_BD_anno, aes(x = x2, xend = x2, y = y1, yend = y2),
               colour = "black") +
  geom_segment(inherit.aes=FALSE, data = iG2rootRH_BD_anno, aes(x = x1, xend = x2, y = y2, yend = y2),
               colour = "black")

iG2rootRH_disper_plot


# G2 rhizo - Flavert
iG2rhizoF_betadisper_G1 <- betadisper(iG2rhizoF_BCdist, group=as.factor(meta_iG2rhizoF$G1_treatment), type = "median")

iG2rhizoF_betadisper_G2 <- betadisper(iG2rhizoF_BCdist, group=as.factor(meta_iG2rhizoF$G2_treatment), type = "median")

iG2rhizoF_betadisper_G1G2 <- betadisper(iG2rhizoF_BCdist, group=as.factor(meta_iG2rhizoF$G1_G2), type = "median")
set.seed(8)
permutest(iG2rhizoF_betadisper_G1, pairwise=TRUE, permutations=999)
set.seed(8)
permutest(iG2rhizoF_betadisper_G2, pairwise=TRUE, permutations=999)
set.seed(8)
permutest(iG2rhizoF_betadisper_G1G2, pairwise=TRUE, permutations=999)
# pairwise:
# C_D vs C_C *
# D_C vs C_C *
# D_D vs C_C **
TukeyHSD(iG2rhizoF_betadisper_G1G2)
#Control_Drought-Control_Control  0.25913598  0.03898235 0.47928960 0.0191238

boxplot(iG2rhizoF_betadisper_G1G2)

# get betadisper data ####
iG2rhizoF_BDdata <- get_betadisper_data(iG2rhizoF_betadisper_G1G2)

iG2rhizoF_BDdata$eigenvalue <- mutate(iG2rhizoF_BDdata$eigenvalue, percent = eig/sum(eig))

iG2rhizoF_BDdata$chull <- group_by(iG2rhizoF_BDdata$eigenvector, group) %>%
  do(data.frame(PCoA1 = .$PCoA1[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])],
                PCoA2 = .$PCoA2[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])])) %>%
  data.frame()

# Now the dataframes are all ready to be completely customisable in ggplot
iG2rhizoF_BD_anno <- data.frame(x1 = c(1, 1, 1), x2 = c(2, 3, 4), 
                                y1 = c(0.2, 0.15, 0.1), y2 = c(0.18, 0.13, 0.08), 
                                xstar = c(1.5, 2, 2.5), ystar = c(0.155, 0.105, 0.055),
                                lab = c("*", "*", "*"))


iG2rhizoF_disper_plot <- ggplot(iG2rhizoF_BDdata$distances, aes(group, distances, col = group)) +
  geom_boxplot(outlier.shape = NA, position = position_dodge(width = 0.55)) +
  geom_point(size=3, shape=21, fill="white") +
  my_theme + labs(title="(FR) G2 Flavert Rhizo") +
  scale_color_manual(name="G1_G2 Consecutive Treatment", values = c("#3399FF","#99CCFF", "#FFCC00", "#FF6600"), labels = c("Control_Control", "Control_Drought", "Drought_Control", "Drought_Drought")) +
  ylab('') + xlab("") +
  labs(caption="G1 x G2 **") + 
  scale_y_continuous(limits = c(0, 0.8))+
  scale_x_discrete(labels = c("C_C", 'C_D', "D_C", "D_D")) +
  theme(legend.position = "none", plot.tag = element_text(face="bold")) + 
  geom_text(inherit.aes=FALSE, data = iG2rhizoF_BD_anno, aes(x = xstar,  y = ystar, label = lab), size=6) +
  geom_segment(inherit.aes=FALSE, data =iG2rhizoF_BD_anno, aes(x = x1, xend = x1, 
           y = y1, yend = y2),
           colour = "black") +
  geom_segment(inherit.aes=FALSE, data = iG2rhizoF_BD_anno, aes(x = x2, xend = x2, 
           y = y1, yend = y2),
           colour = "black") +
  geom_segment(inherit.aes=FALSE, data = iG2rhizoF_BD_anno, aes(x = x1, xend = x2, 
           y = y2, yend = y2),
           colour = "black")
iG2rhizoF_disper_plot


# G2 rhizo - RedHawk
iG2rhizoRH_betadisper_G1 <- betadisper(iG2rhizoRH_BCdist, group=as.factor(meta_iG2rhizoRH$G1_treatment), type = "median")

iG2rhizoRH_betadisper_G2 <- betadisper(iG2rhizoRH_BCdist, group=as.factor(meta_iG2rhizoRH$G2_treatment), type = "median")

iG2rhizoRH_betadisper_G1G2 <- betadisper(iG2rhizoRH_BCdist, group=as.factor(meta_iG2rhizoRH$G1_G2), type = "median")
set.seed(8)
permutest(iG2rhizoRH_betadisper_G1, pairwise=TRUE, permutations=999)
set.seed(8)
permutest(iG2rhizoRH_betadisper_G2, pairwise=TRUE, permutations=999)
set.seed(8)
permutest(iG2rhizoRH_betadisper_G1G2, pairwise=TRUE, permutations=999)
TukeyHSD(iG2rhizoRH_betadisper_G1G2)
# NS

boxplot(iG2rhizoRH_betadisper_G1G2)

# get betadisper data ####
iG2rhizoRH_BDdata <- get_betadisper_data(iG2rhizoRH_betadisper_G1G2)

iG2rhizoRH_BDdata$eigenvalue <- mutate(iG2rhizoRH_BDdata$eigenvalue, percent = eig/sum(eig))

iG2rhizoRH_BDdata$chull <- group_by(iG2rhizoRH_BDdata$eigenvector, group) %>%
  do(data.frame(PCoA1 = .$PCoA1[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])],
                PCoA2 = .$PCoA2[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])])) %>%
  data.frame()

# Now the dataframes are all ready to be completely customisable in ggplot
iG2rhizoRH_disper_plot <- ggplot(iG2rhizoRH_BDdata$distances, aes(group, distances, col = group)) +
  geom_boxplot(outlier.shape = NA, position = position_dodge(width = 0.55)) +
  geom_point(size=3, shape=21, fill="white") +
  my_theme +
  scale_color_manual(name="G1_G2 Consecutive Treatment", values = c("#3399FF","#99CCFF", "#FFCC00", "#FF6600"), labels = c("Control_Control", "Control_Drought", "Drought_Control", "Drought_Drought")) +
  ylab('Distance to Median') + xlab("G1_G2 Consecutive Treatment") +
  scale_x_discrete(labels = c("C_C", 'C_D', "D_C", "D_D")) +
  theme(legend.position = "none")
iG2rhizoRH_disper_plot
```

# example i found to use betadisper in ggplot
```{r}
# functions ####
# getting distances from betadisper() object
betadisper_distances <- function(model){
  temp <- data.frame(group = model$group)
  temp2 <- data.frame(distances = unlist(model$distances))
  temp2$sample <- row.names(temp2)
  temp <- cbind(temp, temp2)
  temp <- dplyr::select(temp, group, sample, dplyr::everything())
  row.names(temp) <- NULL
  return(temp)
}

# getting eigenvalues out of betadisper() object
betadisper_eigenvalue <- function(model){
  temp <- data.frame(eig = unlist(model$eig))
  temp$PCoA <- row.names(temp)
  row.names(temp) <- NULL
  return(temp)
}

# getting the eigenvectors out of a betadisper() object
betadisper_eigenvector <- function(model){
  temp <- data.frame(group = model$group)
  temp2 <- data.frame(unlist(model$vectors))
  temp2$sample <- row.names(temp2)
  temp <- cbind(temp, temp2)
  temp <- dplyr::select(temp, group, sample, dplyr::everything())
  row.names(temp) <- NULL
  return(temp)
}

# get centroids
betadisper_centroids <- function(model){
  temp <- data.frame(unlist(model$centroids))
  temp$group <- row.names(temp)
  temp <- dplyr::select(temp, group, dplyr::everything())
  row.names(temp) <- NULL
  return(temp)
}

# betadisper data
get_betadisper_data <- function(model){
  temp <- list(distances = betadisper_distances(model),
               eigenvalue = betadisper_eigenvalue(model),
               eigenvector = betadisper_eigenvector(model),
               centroids = betadisper_centroids(model))
  return(temp)
}

# get betadisper data ####
betadisper_dat <- get_betadisper_data(mG1root_betadisper)

# do some transformations on the data
betadisper_dat$eigenvalue <- mutate(betadisper_dat$eigenvalue, percent = eig/sum(eig))

# add convex hull points ####
# this could be put in a function
betadisper_dat$chull <- group_by(betadisper_dat$eigenvector, group) %>%
  do(data.frame(PCoA1 = .$PCoA1[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])],
                PCoA2 = .$PCoA2[c(chull(.$PCoA1, .$PCoA2), chull(.$PCoA1, .$PCoA2)[1])])) %>%
  data.frame()

# combine centroid and eigenvector dataframes for plotting
betadisper_lines <- merge(select(betadisper_dat$centroids, group, PCoA1, PCoA2), select(betadisper_dat$eigenvector, group, PCoA1, PCoA2), by = c('group'))

# Now the dataframes are all ready to be completely customisable in ggplot
# plot betadispersion plot
ggplot() +
  geom_point(aes(PCoA1, PCoA2, col = group), betadisper_dat$centroids, size = 4) +
  geom_point(aes(PCoA1, PCoA2, col = group), betadisper_dat$eigenvector) +
  geom_path(aes(PCoA1, PCoA2, col = group, group = group), betadisper_dat$chull ) +
  geom_segment(aes(x = PCoA1.x, y = PCoA2.x, yend = PCoA2.y, xend = PCoA1.y, group = row.names(betadisper_lines), col = group), betadisper_lines) +
  theme_bw(base_size = 12, base_family = 'Helvetica') +
  ylab('PCoA Axis 2 [25.4%]') +
  xlab('PCoA Axis 2 [53.8%]') +
  scale_color_manual('', values = c('black', 'grey'), labels = c("Control", 'Drought')) +
  theme(legend.position = c(0.9, 0.3)) #+
  #coord_fixed(sqrt(d_PCA_sum$percent[2]/d_PCA_sum$percent[1]))

# plot distances from centroid
ggplot(betadisper_dat$distances, aes(group, distances, col = group)) +
  geom_boxplot(outlier.shape = NA, width = 0.5, position = position_dodge(width = 0.55)) +
  stat_summary(geom = 'crossbar', color = 'red', width = 0.5, fun.data = function(x){ return(c(y=median(x), ymin=median(x), ymax=median(x)))}) +
  geom_point(size=2) +
  my_theme +
  scale_color_manual(name="G1 Treatment", values = c('black', 'grey'), labels = c("Control", 'Drought')) +
  scale_fill_manual(name="G1 Treatment", values = c('black', 'grey'), labels = c("Control", 'Drought')) +
  ylab('Distance to Median') +
  scale_x_discrete(labels = c("Control", 'Drought'))

# okay so this works, it's very complicated but once you have the functions it's pretty easy to make all the pieces for each dataset. Will be a little clunky to repeat this for each graph but oh well.
```

# relative abundance barplots
```{r}
root_16S_gen1
root_16S_gen2
rhizo_16S_gen1
rhizo_16S_gen2


# G1 roots
data.frame(tax_table(root_16S_gen1))
data.frame(sample_data(root_16S_gen1))

# group by Family
root_g1_family <- tax_glom(root_16S_gen1, taxrank = "Family", NArm = F)
root_g1_family.ra <- transform_sample_counts(root_g1_family, function(x) x/sum(x))
root_g1_family.ra
sample_data(root_g1_family.ra)
data.frame(otu_table(root_g1_family.ra))
fam_tax <- data.frame(tax_table(root_g1_family.ra))
unique(fam_tax$Order)

df.root_g1_family <- psmelt(root_g1_family.ra) 
df.root_g1_family$Family <- gsub("f__", "", df.root_g1_family$Family)
df.root_g1_family$Family <- gsub("Mitochondria", "Rickettsiales", df.root_g1_family$Family)

df.root_g1_family_rhizobiacaea <- filter(df.root_g1_family, Family == "Rhizobiaceae")
aov_g1_root_rhizobia <- aov(Abundance ~ G1_treatment*Group, data=df.root_g1_family_rhizobiacaea)
summary(aov_g1_root_rhizobia)
#                    Df Sum Sq Mean Sq F value Pr(>F)  
# G1_treatment        1  0.001 0.00141   0.016  0.900  
# Group               2  0.568 0.28383   3.214  0.048 *
# G1_treatment:Group  2  0.015 0.00741   0.084  0.920  
# Residuals          54  4.769 0.08832  
tukey_g1_root_rhizobia <- tukey_hsd(aov_g1_root_rhizobia)
tukey_g1_root_rhizobia
#	Group	INRAE_Red_Hawk	MSU_Red_Hawk	0	-0.236956567	-0.46344512	-0.01046801	0.0384	*


df.root_g1_family <- df.root_g1_family %>%
  summarise(Mean = mean(Abundance), .by = c(Group, G1_treatment, Family)) %>%
  arrange(-Mean)  

df.root_g1_family$Family <- as.character(df.root_g1_family$Family) #convert to character

#simple way to rename phyla with < 1% abundance
df.root_g1_family$Family[df.root_g1_family$Mean < 0.005] <- "< 0.5% abund."

#Count # phyla to set color palette
Count = length(unique(df.root_g1_family$Family))
Count



# New facet label names for plant variable
treatment.labs <- c("G1 Control", "G1 Drought")
names(treatment.labs) <- c("Control", "Drought")

# Create the plot

colors_a <- paletteer_d("ggthemes::Classic_20")
colors_b <- paletteer_d("tvthemes::kimPossible") #12

colors_22 <- c(colors_a[1:14], colors_b[11], colors_b[12], colors_a[16:20])


plot.root_g1_family <- ggplot(data=df.root_g1_family, aes(x=G1_treatment, y=Mean, fill=Family)) + 
                      geom_bar(aes(), stat="identity", position="fill") + 
                      scale_fill_manual(values=colors_22) +
                      facet_wrap(vars(Group), nrow=1)
plot.root_g1_family



# group by Order
root_g1_order <- tax_glom(root_16S_gen1, taxrank = "Order", NArm = T)
root_g1_order.ra <- transform_sample_counts(root_g1_order, function(x) x/sum(x))
root_g1_order.ra
sample_data(root_g1_order.ra)
data.frame(otu_table(root_g1_order.ra))
data.frame(tax_table(root_g1_order.ra))

df.root_g1_order <- psmelt(root_g1_order.ra) 
df.root_g1_order$Order <- gsub("o__", "", df.root_g1_order$Order)
df.root_g1_order$Order <- gsub("Chloroplast", "c__Oxyphotobacteria", df.root_g1_order$Order)

df.root_g1_order <- df.root_g1_order %>%
  summarise(Mean = mean(Abundance), .by = c(Group, G1_treatment, Order)) %>%
  arrange(-Mean)  

df.root_g1_order$Order <- as.character(df.root_g1_order$Order) #convert to character

#simple way to rename phyla with < 1% abundance
df.root_g1_order$Order[df.root_g1_order$Mean < 0.005] <- "< 0.5% abund."

#Count # phyla to set color palette
length(unique(df.root_g1_order$Order))




# New facet label names for plant variable
treatment.labs <- c("G1 Control", "G1 Drought")
names(treatment.labs) <- c("Control", "Drought")

# Create the plot

plot.root_g1_order <- ggplot(data=df.root_g1_order, aes(x=G1_treatment, y=Mean, fill=Order)) + 
                     geom_bar(aes(), stat="identity", position="fill") + facet_wrap(vars(Group), nrow=1)
plot.root_g1_order
plot.root_g1_family



#### keep generations together ####
all_16S_gen1
all_16S_gen2

# G1 group by Family
all_g1_family <- tax_glom(all_16S_gen1, taxrank = "Family", NArm = F)
all_g1_family.ra <- transform_sample_counts(all_g1_family, function(x) x/sum(x))
all_g1_family.ra
sample_data(all_g1_family.ra)
data.frame(otu_table(all_g1_family.ra))
fam_tax_all <- data.frame(tax_table(all_g1_family.ra))
unique(fam_tax_all$Family)

df.all_g1_family <- psmelt(all_g1_family.ra) 
df.all_g1_family$Family <- gsub("f__", "", df.all_g1_family$Family)
df.all_g1_family$Family <- gsub("Mitochondria", "Rickettsiales", df.all_g1_family$Family)

# stats on rhizobiaceae
df.g1_family_rhizobiacaea <- filter(df.all_g1_family, Family == "Rhizobiaceae")
aov_g1_rhizobia <- aov(Abundance ~ G1_treatment*Compartment*Group, data=df.g1_family_rhizobiacaea)
summary(aov_g1_rhizobia)
#                                 Df Sum Sq Mean Sq F value   Pr(>F)    
# G1_treatment                     1  0.005  0.0047   0.095   0.7586    
# Compartment                      1  1.264  1.2642  25.613 1.72e-06 ***
# Group                            2  0.374  0.1868   3.784   0.0258 *  
# G1_treatment:Compartment         1  0.015  0.0148   0.299   0.5856    
# G1_treatment:Group               2  0.023  0.0114   0.230   0.7947    
# Compartment:Group                2  0.213  0.1064   2.156   0.1207    
# G1_treatment:Compartment:Group   2  0.002  0.0010   0.020   0.9805    
# Residuals                      108  5.331  0.0494                     
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
tukey_g1_rhizobia <- tukey_hsd(aov_g1_rhizobia)
tukey_g1_rhizobia
df.g1_family_rhizo <- filter(df.g1_family_rhizobiacaea, Compartment == "Rhizosphere")
df.g1_family_rhizoF <- filter(df.g1_family_rhizo, Group == "INRAE_Flavert")
df.g1_family_rhizoRH <- filter(df.g1_family_rhizo, Group == "INRAE_Red_Hawk")
aov_rhizoF <- aov(Abundance ~ G1_treatment, data=df.g1_family_rhizoF)
summary(aov_rhizoF) #NS
aov_rhizoRH <- aov(Abundance ~ G1_treatment, data=df.g1_family_rhizoRH)
summary(aov_rhizoRH) #NS

df.all_g1_family <- df.all_g1_family %>%
  summarise(Mean = mean(Abundance), .by = c(Group, Compartment, G1_treatment, Family)) %>%
  arrange(-Mean)  

df.all_g1_family$Family <- as.character(df.all_g1_family$Family) #convert to character

#simple way to rename phyla with < 1% abundance
df.all_g1_family$Family[df.all_g1_family$Mean < 0.01] <- "< 1% abund."

#Count # phyla to set color palette
length(unique(df.all_g1_family$Family))


# New facet label names for plant variable
group.labs <- c("FRANCE_Flavert", "FRANCE_Red_Hawk", "USA_Red_Hawk")
names(group.labs) <- c("INRAE_Flavert", "INRAE_Red_Hawk", "MSU_Red_Hawk")

# Create the plot

colors_a <- paletteer_d("ggthemes::Classic_20")
colors_b <- paletteer_d("tvthemes::kimPossible") #12
colors_c <- paletteer_dynamic("cartography::multi.pal", 20)

colors_27 <- c(colors_a[1:14], colors_a[16:20], colors_b[12], colors_b[11], colors_b[2:4], colors_b[7:8], colors_c[1])


plot.all_g1_family <- ggplot(data=df.all_g1_family, aes(x=G1_treatment, y=Mean, fill=Family, color=Family)) + 
                      geom_bar(aes(), stat="identity", position="fill") + 
                      scale_fill_manual(values=colors_27) +
                        scale_color_manual(values=colors_27) +
                      facet_grid(rows=vars(Compartment), cols=(vars(Group)), labeller = labeller(Group=group.labs)) + 
                     guides(fill=guide_legend(ncol=1))+
                     labs(y= "Mean Relative Abundance", x="G1 Treatment")
plot.all_g1_family



# G2 group by Family
all_g2_family <- tax_glom(all_16S_gen2, taxrank = "Family", NArm = F)
all_g2_family.ra <- transform_sample_counts(all_g2_family, function(x) x/sum(x))
all_g2_family.ra
sample_data(all_g2_family.ra)
data.frame(otu_table(all_g2_family.ra))
fam_tax_all_g2 <- data.frame(tax_table(all_g2_family.ra))
unique(fam_tax_all_g2$Family)

df.all_g2_family <- psmelt(all_g2_family.ra) 
df.all_g2_family$Family <- gsub("f__", "", df.all_g2_family$Family)
df.all_g2_family$Family <- gsub("Mitochondria", "Rickettsiales", df.all_g2_family$Family)

# stats on rhizobiaceae
df.g2_family_rhizobiacaea <- filter(df.all_g2_family, Family == "Rhizobiaceae")
df.g2_family_rhizo <- filter(df.g2_family_rhizobiacaea, Compartment == "Rhizosphere")
df.g2_family_rhizoF <- filter(df.g2_family_rhizo, Group == "INRAE_Flavert")
df.g2_family_rhizoRH <- filter(df.g2_family_rhizo, Group == "INRAE_Red_Hawk")
aov_g2rhizoF <- aov(Abundance ~ G1_G2, data=df.g2_family_rhizoF)
summary(aov_g2rhizoF) #NS
tukey_hsd(aov_g2rhizoF)
aov_g2rhizoRH <- aov(Abundance ~ G1_G2, data=df.g2_family_rhizoRH)
summary(aov_g2rhizoRH) #NS
tukey_hsd(aov_g2rhizoRH)

df.all_g2_family <- df.all_g2_family %>%
  summarise(Mean = mean(Abundance), .by = c(Group, Compartment, G1_G2, Family)) %>%
  arrange(-Mean)  

df.all_g2_family$Family <- as.character(df.all_g2_family$Family) #convert to character

#simple way to rename phyla with < 1% abundance
df.all_g2_family$Family[df.all_g2_family$Mean < 0.01] <- "< 1% abund."

#Count # phyla to set color palette
length(unique(df.all_g2_family$Family))


# New facet label names for plant variable
group.labs <- c("FRANCE_Flavert", "FRANCE_Red_Hawk", "USA_Red_Hawk")
names(group.labs) <- c("INRAE_Flavert", "INRAE_Red_Hawk", "MSU_Red_Hawk")

# Create the plot

colors_a <- paletteer_d("ggthemes::Classic_20")
colors_b <- paletteer_d("tvthemes::kimPossible") #12
colors_c <- paletteer_dynamic("cartography::multi.pal", 20)

colors_34 <- c("#1F77B4FF", "#AEC7E8FF", "#FF7F0EFF", "#FFBB78FF", "#2CA02CFF", "#98DF8AFF", "#D62728FF", "#FF9896FF", "#9467BDFF", "#C5B0D5FF", "#6B42C8FF", "#8C564BFF", "#C49C94FF", "#E377C2FF", "#C9D73DFF", "#C555CBFF", "#502E71FF", "#C49A3FFF", "#DBDB8DFF", "#D7652DFF", "#CF4C8BFF",  "#17BECFFF", "#9EDAE5FF", "#000000FF", "#5D8D9CFF",  "#C8B693FF", "#FFF9F5FF", "#024663FF", "#70D423FF", "#891604FF", "#FFCE45FF", "#5D734DFF", "#CB7C77FF")


plot.all_g2_family <- ggplot(data=df.all_g2_family, aes(x=G1_G2, y=Mean, fill=Family, color=Family)) + 
                      geom_bar(aes(), stat="identity", position="fill") + 
                      scale_fill_manual(values=colors_34) +
                      scale_color_manual(values=colors_34) +
                      facet_grid(rows=vars(Compartment), cols=(vars(Group)), labeller = labeller(Group=group.labs)) + 
                     guides(fill=guide_legend(ncol=1))+
                     labs(y= "Mean Relative Abundance", x="G1_G2 Treatment") +
                      theme(axis.text.x = element_text(angle = 45, hjust=1))
plot.all_g2_family
plot.all_g1_family


ggsave(filename="plot.all_g1_family.pdf", plot=plot.all_g1_family, path="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/", width = 12, height =8.5, units= "in", dpi=300)

ggsave(filename="plot.all_g2_family.pdf", plot=plot.all_g2_family, path="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/", width = 12, height =9.5, units= "in", dpi=300)
```




# Trying standardization for alpha diversity with decostand
```{r}
# decostand "normalize" method
asv_table <- as.matrix(otu_table(all_16S_phyloseq))

asv_table <- t(asv_table)

asv_stand <- decostand(asv_table, "normalize")

asv_stand_phyloseq <- otu_table(asv_stand, taxa_are_rows = FALSE)

meta <- sample_data(all_16S_phyloseq)
tax <- tax_table(all_16S_phyloseq)


standardized_richness <- c(rowSums(asv_stand))

stand_rich <- as.data.frame(standardized_richness)
stand_rich <- cbind(rownames(stand_rich), stand_rich) 
colnames(stand_rich)[1] <- "SampleID"
stand_rich

asv_stand <- as.data.frame(asv_stand)
asv_stand <- cbind(rownames(asv_stand), asv_stand) 
colnames(asv_stand)[1] <- "SampleID"
asv_stand

table_w_rowsums <- left_join(asv_stand, stand_rich)

metadata <- data.frame(meta)

meta_w_rowsums <- left_join(metadata, stand_rich)
meta_w_rowsums[126]

# "standardized_richness" was added at end of metadata table

stand_richness_plot <- ggplot(data = meta_w_rowsums, aes(x=Generation, y=standardized_richness, color=Treatment)) + geom_boxplot() + facet_grid(rows=vars(Compartment), cols=vars(Group), scales="free")

stand_richness_plot



# decostand "standardize" method
asv_table <- as.matrix(otu_table(all_16S_phyloseq))

asv_table <- t(asv_table)

asv_stand2 <- decostand(asv_table, "pa")

group = c(rownames(asv_stand2))

standardized=_richness2 <- rowSums(asv_stand2, na.rm=TRUE)

z_scores <- (standardized_richness2-mean(standardized_richness2))/sd(standardized_richness2)

stand_rich2 <- as.data.frame(z_scores)
stand_rich2 <- cbind(rownames(stand_rich2), stand_rich2) 
colnames(stand_rich2)[1] <- "SampleID"
stand_rich2

asv_stand2 <- as.data.frame(asv_stand2)
asv_stand2 <- cbind(rownames(asv_stand2), asv_stand2) 
colnames(asv_stand2)[1] <- "SampleID"
asv_stand2

metadata <- data.frame(meta)

meta_w_rowsums2 <- left_join(metadata, stand_rich2)
meta_w_rowsums2[126]

# "standardized_richness" was added at end of metadata table

stand_richness_plot2 <- ggplot(data = meta_w_rowsums2, aes(x=Generation, y=z_scores, color=Treatment)) + geom_boxplot() + facet_grid(rows=vars(Compartment), cols=vars(Group), scales="free")

stand_richness_plot2
```

# indicator species analysis
I will use the "indicspecies" package which requires a matrix (OTU table) with samples/sites in rows and species in columns, we also need a vector containing the "group" classifications. Vector needs to be the same length as the number of samples in the matrix.
```{r indicspecies MSU}
# Try for MSU dataset, group by generation, G1_treatment, G2_treatment, G1_G2

MSU_full_ASV_table <- t(otu_table(MSU_phyloseq))

MSU_full_ASV_table <- as.data.frame(MSU_full_ASV_table)

MSU_full_metadata <- data.frame(sample_data(MSU_phyloseq))

msu_generation <- c(MSU_full_metadata$Generation)
msu_compartment <- c(MSU_full_metadata$Compartment)
msu_G1_treatment <- c(MSU_full_metadata$G1_treatment)
msu_G2_treatment <- c(MSU_full_metadata$G2_treatment)
msu_G1_G2 <- c(MSU_full_metadata$G1_G2)

indic_msu_generation <- multipatt(MSU_full_ASV_table, msu_generation, duleg=TRUE, control=how(nperm=999))

summary(indic_msu_generation)


# Generation 1

MSU_G1_ASV <- as.data.frame(t(otu_table(MSU_gen1)))
MSU_G1_metadata <- data.frame(sample_data(MSU_gen1))
msu_g1_trt <- c(MSU_G1_metadata$G1_treatment)
msu_g1_compartment <- c(MSU_G1_metadata$Compartment)

indic_msu_gen1_trt <- multipatt(MSU_G1_ASV, msu_g1_trt, duleg=TRUE, control=how(nperm=999))
summary(indic_msu_gen1_trt)
# G1 control group: 9 species
# G1 drought group: 31 species

indic_msu_gen1_compart <- multipatt(MSU_G1_ASV, msu_g1_compartment, duleg=TRUE, control=how(nperm=999))
summary(indic_msu_gen1_compart)
# 700+ species based on compartment, look at treatments within compartment

mG1root_ASV <- as.data.frame(t(otu_table(MSU_gen1_root)))
mG1root_trt <- c(meta_mG1root$G1_treatment)

mG1rhizo_ASV <- as.data.frame(t(otu_table(MSU_gen1_rhizo)))
mG1rhizo_trt <- c(meta_mG1rhizo$G1_treatment)

indic_mG1root <- multipatt(mG1root_ASV, mG1root_trt, duleg=TRUE, control=how(nperm=999))
summary(indic_mG1root)
# 8 control, 21 drought
sink(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_mG1root.csv")
summary(indic_mG1root)
sink()

indic_mG1rhizo <- multipatt(mG1rhizo_ASV, mG1rhizo_trt, duleg=TRUE, control=how(nperm=999))
summary(indic_mG1rhizo)
# 11 control, 10 drought
sink(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_mG1rhizo.csv")
summary(indic_mG1rhizo)
sink()


# G2 test G2_treatment and G1_G2
mG2root_ASV <- as.data.frame(t(otu_table(MSU_gen2_root)))
mG2root_trt2 <- c(meta_mG2root$G2_treatment)
mG2root_trt1 <- c(meta_mG2root$G1_treatment)
mG2root_g1g2 <- c(meta_mG2root$G1_G2)

mG2rhizo_ASV <- as.data.frame(t(otu_table(MSU_gen2_rhizo)))
mG2rhizo_trt2 <- c(meta_mG2rhizo$G2_treatment)
mG2rhizo_trt1 <- c(meta_mG2rhizo$G1_treatment)
mG2rhizo_g1g2 <- c(meta_mG2rhizo$G1_G2)

indic_mG2root_trt2 <- multipatt(mG2root_ASV, mG2root_trt2, duleg=TRUE, control=how(nperm=999))
summary(indic_mG2root_trt2)
# 112 control, 43 drought
sink(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_mG2root_trt2.csv")
summary(indic_mG2root_trt2)
sink()

indic_mG2root_trt1 <- multipatt(mG2root_ASV, mG2root_trt1, duleg=TRUE, control=how(nperm=999))
summary(indic_mG2root_trt1)
# 12 control, 48 drought
sink(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_mG2root_trt1.csv")
summary(indic_mG2root_trt1)
sink()

indic_mG2root_g1g2 <- multipatt(mG2root_ASV, mG2root_g1g2, duleg=FALSE, control=how(nperm=999))
summary(indic_mG2root_g1g2)
 # Selected number of species: 170 
 # Number of species associated to 1 group: 124 
 # Number of species associated to 2 groups: 28 
 # Number of species associated to 3 groups: 18 
sink(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_mG2root_g1g2.csv")
summary(indic_mG2root_g1g2)
sink()



indic_mG2rhizo_trt2 <- multipatt(mG2rhizo_ASV, mG2rhizo_trt2, duleg=TRUE, control=how(nperm=999))
summary(indic_mG2rhizo_trt2)
# 53 control, 67 drought

sink(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_mG2rhizo_trt2.csv")
summary(indic_mG2rhizo_trt2)
sink()


indic_mG2rhizo_trt1 <- multipatt(mG2rhizo_ASV, mG2rhizo_trt1, duleg=TRUE, control=how(nperm=999))
summary(indic_mG2rhizo_trt1)
sink(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_mG2rhizo_trt1.csv")
summary(indic_mG2rhizo_trt1)
sink()



indic_mG2rhizo_g1g2 <- multipatt(mG2rhizo_ASV, mG2rhizo_g1g2, duleg=FALSE, control=how(nperm=999))
summary(indic_mG2rhizo_g1g2)
 # Selected number of species: 160 
 # Number of species associated to 1 group: 115 
 # Number of species associated to 2 groups: 33 
 # Number of species associated to 3 groups: 12 

sink(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_mG2rhizo_g1g2.csv")
summary(indic_mG2rhizo_g1g2)
sink()

```


# INRAE indicspecies
```{r indicspecies INRAE}
# Generation 1
INRAE_G1_ASV <- as.data.frame(otu_table(INRAE_gen1))
INRAE_G1_meta <- data.frame(sample_data(INRAE_gen1))
ig1_genotype <- c(INRAE_G1_meta$Genotype)

indic_ig1_geno <- multipatt(INRAE_G1_ASV, ig1_genotype, duleg=TRUE, control=how(nperm=999))
summary(indic_ig1_geno)
# Fl 53 species
# RH 25 species

### G1 FLAVERT
iG1rootF_ASV <- as.data.frame(otu_table(INRAE_gen1_rootF))
iG1rootF_trt <- c(meta_iG1rootF$G1_treatment)

iG1rhizoF_ASV <- as.data.frame(otu_table(INRAE_gen1_rhizoF))
iG1rhizoF_trt <- c(meta_iG1rhizoF$G1_treatment)

indic_iG1rootF <- multipatt(iG1rootF_ASV, iG1rootF_trt, duleg=TRUE, control=how(nperm=999))
summary(indic_iG1rootF)
# 28 in drought, 0 in control
sink(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_iG1rootF.csv")
summary(indic_iG1rootF)
sink()

indic_iG1rhizoF <- multipatt(iG1rhizoF_ASV, iG1rhizoF_trt, duleg=TRUE, control=how(nperm=999))
summary(indic_iG1rhizoF)
# 5 drought, 28 control
sink(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_iG1rhizoF.csv")
summary(indic_iG1rhizoF)
sink()


### G1 REDHAWK
iG1rootRH_ASV <- as.data.frame(otu_table(INRAE_gen1_rootRH))
iG1rootRH_trt <- c(meta_iG1rootRH$G1_treatment)

iG1rhizoRH_ASV <- as.data.frame(otu_table(INRAE_gen1_rhizoRH))
iG1rhizoRH_trt <- c(meta_iG1rhizoRH$G1_treatment)

indic_iG1rootRH <- multipatt(iG1rootRH_ASV, iG1rootRH_trt, duleg=TRUE, control=how(nperm=999))
summary(indic_iG1rootRH)
# 3 in drought, 3 in control
sink(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_iG1rootRH.csv")
summary(indic_iG1rootRH)
sink()

indic_iG1rhizoRH <- multipatt(iG1rhizoRH_ASV, iG1rhizoRH_trt, duleg=TRUE, control=how(nperm=999))
summary(indic_iG1rhizoRH)
# 50 drought, 1 control
sink(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_iG1rhizoRH.csv")
summary(indic_iG1rhizoRH)
sink()


### G2 FLAVERT
iG2rootF_ASV <- as.data.frame(otu_table(INRAE_gen2_rootF))
iG2rootF_trt2 <- c(meta_iG2rootF$G2_treatment)
iG2rootF_g1g2 <- c(meta_iG2rootF$G1_G2)
iG2rootF_trt1 <- c(meta_iG2rootF$G1_treatment)

iG2rhizoF_ASV <- as.data.frame(otu_table(INRAE_gen2_rhizoF))
iG2rhizoF_trt2 <- c(meta_iG2rhizoF$G2_treatment)
iG2rhizoF_g1g2 <- c(meta_iG2rhizoF$G1_G2)
iG2rhizoF_trt1 <- c(meta_iG2rhizoF$G1_treatment)

indic_iG2rootF_trt2 <- multipatt(iG2rootF_ASV, iG2rootF_trt2, duleg=TRUE, control=how(nperm=999))
summary(indic_iG2rootF_trt2)
# 2 in drought, 1 in control
sink(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_iG2rootF_trt2.csv")
summary(indic_iG2rootF_trt2)
sink()

indic_iG2rootF_trt1 <- multipatt(iG2rootF_ASV, iG2rootF_trt1, duleg=TRUE, control=how(nperm=999))
summary(indic_iG2rootF_trt1)
# 0 taxa

indic_iG2rootF_g1g2 <- multipatt(iG2rootF_ASV, iG2rootF_g1g2, duleg=FALSE, control=how(nperm=999))
summary(indic_iG2rootF_g1g2)
 # Total number of species: 113
 # Selected number of species: 5 
 # Number of species associated to 1 group: 2 
 # Number of species associated to 2 groups: 3 
 # Number of species associated to 3 groups: 0 
sink(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_iG2rootF_g1g2.csv")
summary(indic_iG2rootF_g1g2)
sink()

indic_iG2rhizoF_trt2 <- multipatt(iG2rhizoF_ASV, iG2rhizoF_trt2, duleg=TRUE, control=how(nperm=999))
summary(indic_iG2rhizoF_trt2)
# 0 drought, 17 control
sink(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_iG2rhizoF_trt2.csv")
summary(indic_iG2rhizoF_trt2)
sink()

indic_iG2rhizoF_trt1 <- multipatt(iG2rhizoF_ASV, iG2rhizoF_trt1, duleg=TRUE, control=how(nperm=999))
summary(indic_iG2rhizoF_trt1)
# 120 control, 0 drought
sink(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_iG2rhizoF_trt1.csv")
summary(indic_iG2rhizoF_trt1)
sink()

indic_iG2rhizoF_g1g2 <- multipatt(iG2rhizoF_ASV, iG2rhizoF_g1g2, duleg=FALSE, control=how(nperm=999))
summary(indic_iG2rhizoF_g1g2)
 # Total number of species: 602
 # Selected number of species: 95 
 # Number of species associated to 1 group: 66 
 # Number of species associated to 2 groups: 24 
 # Number of species associated to 3 groups: 5 
sink(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_iG2rhizoF_g1g2.csv")
summary(indic_iG2rhizoF_g1g2)
sink()



### G2 REDHAWK
iG2rootRH_ASV <- as.data.frame(otu_table(INRAE_gen2_rootRH))
iG2rootRH_trt2 <- c(meta_iG2rootRH$G2_treatment)
iG2rootRH_g1g2 <- c(meta_iG2rootRH$G1_G2)
iG2rootRH_trt1 <- c(meta_iG2rootRH$G1_treatment)

iG2rhizoRH_ASV <- as.data.frame(otu_table(INRAE_gen2_rhizoRH))
iG2rhizoRH_trt2 <- c(meta_iG2rhizoRH$G2_treatment)
iG2rhizoRH_g1g2 <- c(meta_iG2rhizoRH$G1_G2)
iG2rhizoRH_trt1 <- c(meta_iG2rhizoRH$G1_treatment)

indic_iG2rootRH_trt2 <- multipatt(iG2rootRH_ASV, iG2rootRH_trt2, duleg=TRUE, control=how(nperm=999))
summary(indic_iG2rootRH_trt2)
# 0 indicator taxa

indic_iG2rootRH_trt1 <- multipatt(iG2rootRH_ASV, iG2rootRH_trt1, duleg=TRUE, control=how(nperm=999))
summary(indic_iG2rootRH_trt1)
# 0 indicator taxa

indic_iG2rootRH_g1g2 <- multipatt(iG2rootRH_ASV, iG2rootRH_g1g2, duleg=FALSE, control=how(nperm=999))
summary(indic_iG2rootRH_g1g2)
 # 0 taxa

indic_iG2rhizoRH_trt2 <- multipatt(iG2rhizoRH_ASV, iG2rhizoRH_trt2, duleg=TRUE, control=how(nperm=999))
summary(indic_iG2rhizoRH_trt2)
# 2 drought, 3 control
sink(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_iG2rhizoRH_trt2.csv")
summary(indic_iG2rhizoRH_trt2)
sink()

indic_iG2rhizoRH_trt1 <- multipatt(iG2rhizoRH_ASV, iG2rhizoRH_trt1, duleg=TRUE, control=how(nperm=999))
summary(indic_iG2rhizoRH_trt1)
# 5 control, 0 drought
sink(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_iG2rhizoRH_trt1.csv")
summary(indic_iG2rhizoRH_trt1)
sink()

indic_iG2rhizoRH_g1g2 <- multipatt(iG2rhizoRH_ASV, iG2rhizoRH_g1g2, duleg=FALSE, control=how(nperm=999))
summary(indic_iG2rhizoRH_g1g2)
 # Total number of species: 397
 # Selected number of species: 5 
 # Number of species associated to 1 group: 4 
 # Number of species associated to 2 groups: 0 
 # Number of species associated to 3 groups: 1 
sink(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_iG2rhizoRH_g1g2.csv")
summary(indic_iG2rhizoRH_g1g2)
sink()


write.csv(data.frame(tax_table(INRAE_root_rhizo_CD)), file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/INRAE_tax.csv")

write.csv(data.frame(tax_table(MSU_phyloseq)), file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/MSU_tax.csv")



```

# Merge indicspecies with tax table
```{r indic_tax}
MSU_TAX <- read.csv("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/MSU_tax.csv")

setwd("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/")

df_mG2rhizo_g1g2 <- read.csv("indic_mG2rhizo_g1g2.csv")
tax_mG2rhizo_g1g2 <- left_join(df_mG2rhizo_g1g2, MSU_TAX, by="OTUID")
write.csv(tax_mG2rhizo_g1g2, "tax_mG2rhizo_g1g2.csv")

df_mG2rhizo_trt2 <- read.csv("indic_mG2rhizo_trt2.csv")
tax_mG2rhizo_trt2 <- left_join(df_mG2rhizo_trt2, MSU_TAX, by="OTUID")
write.csv(tax_mG2rhizo_trt2, "tax_mG2rhizo_trt2.csv")

df_mG2rhizo_trt1 <- read.csv("indic_mG2rhizo_trt1.csv")
tax_mG2rhizo_trt1 <- left_join(df_mG2rhizo_trt1, MSU_TAX, by="OTUID")
write.csv(tax_mG2rhizo_trt1, "tax_mG2rhizo_trt1.csv")

df_mG2root_g1g2 <- read.csv("indic_mG2root_g1g2.csv")
tax_mG2root_g1g2 <- left_join(df_mG2root_g1g2, MSU_TAX, by="OTUID")
write.csv(tax_mG2root_g1g2, "tax_mG2root_g1g2.csv")

df_mG2root_trt2 <- read.csv("indic_mG2root_trt2.csv")
tax_mG2root_trt2 <- left_join(df_mG2root_trt2, MSU_TAX, by="OTUID")
write.csv(tax_mG2root_trt2, "tax_mG2root_trt2.csv")

df_mG2root_trt1 <- read.csv("indic_mG2root_trt1.csv")
tax_mG2root_trt1 <- left_join(df_mG2root_trt1, MSU_TAX, by="OTUID")
write.csv(tax_mG2root_trt1, "tax_mG2root_trt1.csv")

df_mG1root <- read.csv("indic_mG1root.csv")
tax_mG1root <- left_join(df_mG1root, MSU_TAX, by="OTUID")
write.csv(tax_mG1root, "tax_mG1root.csv")

df_mG1rhizo <- read.csv("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_mG1rhizo.csv")
tax_mG1rhizo <- left_join(df_mG1rhizo, MSU_TAX, by="OTUID")
write.csv(tax_mG1rhizo, "/Users/Abby/OneDrive - Michigan State University/multigen_analysis/tax_mG1rhizo.csv")




mG2rhizo_full_indictaxa <- read.csv("indic_taxa_mG2rhizo_full.csv")
mG2rhizo_taxa_plot <- ggplot(data=mG2rhizo_full_indictaxa, aes(x=Group, fill=Class)) + 
  geom_bar(position="stack") + 
  xlab("Treatment Group") + ylab("Number of Indicator Taxa") + labs(title="USA G2 Rhizosphere Indicator Taxa")
mG2rhizo_taxa_plot


mG2root_full_indictaxa <- read.csv("indic_taxa_mG2root_full.csv")
mG2root_taxa_plot <- ggplot(data=mG2root_full_indictaxa, aes(x=Group, fill=Class)) + 
  geom_bar(position="stack") + 
  xlab("Treatment Group") + ylab("Number of Indicator Taxa") + labs(title="USA G2 Root Indicator Taxa")
mG2root_taxa_plot


mG1root_taxa_plot <- ggplot(data=tax_mG1root, aes(x=Group, fill=Order)) + 
  geom_bar(position="stack") + 
  xlab("Treatment Group") + ylab("Number of Indicator Taxa") + labs(title="USA G1 Root Indicator Taxa")
mG1root_taxa_plot


tax_mG1rhizo_fix <- read.csv("tax_mG1rhizo.csv")
mG1rhizo_taxa_plot <- ggplot(data=tax_mG1rhizo_fix, aes(x=Group, fill=Order)) + 
  geom_bar(position="stack") + 
  xlab("Treatment Group") + ylab("Number of Indicator Taxa") + labs(title="USA G1 Rhizosphere Indicator Taxa")
mG1rhizo_taxa_plot

MSU_G1indicator_taxa_plots <- mG1root_taxa_plot / mG1rhizo_taxa_plot

MSU_G2indicator_taxa_plots <- mG2root_taxa_plot / mG2rhizo_taxa_plot

ggsave(plot=MSU_G1indicator_taxa_plots, "MSU_G1indicator_taxa_plots.png", width=8.5, height=11, units="in")

ggsave(plot=MSU_G2indicator_taxa_plots, "MSU_G2indicator_taxa_plots.png", width=15, height=11, units="in")




##### FRANCE #####

INRAE_tax <- read.csv("INRAE_tax.csv")

# FLAVERT #

# Generation 1
df_iG1rootF <- read.csv("indic_iG1rootF.csv")
tax_iG1rootF <- left_join(df_iG1rootF, INRAE_tax, by="OTUID")
write.csv(tax_iG1rootF, "tax_iG1rootF.csv")

df_iG1rhizoF <- read.csv("indic_iG1rhizoF.csv")
tax_iG1rhizoF <- left_join(df_iG1rhizoF, INRAE_tax, by="OTUID")
write.csv(tax_iG1rhizoF, "tax_iG1rhizoF.csv")

tax_iG1root.rhizoF_fix <- read.csv("tax_iG1root.rhizoF_full.csv")
iG1root.rhizoF_taxa_plot <- ggplot(data=tax_iG1root.rhizoF_fix, aes(x=Group, fill=Order)) + 
  geom_bar(position="stack") + 
  xlab("Treatment Group") + ylab("Number of Indicator Taxa") + labs(title="France G1 Flavert Indicator Taxa") + 
  facet_wrap(~Compartment, scales="free")
iG1root.rhizoF_taxa_plot

# Gen 2
df_iG2rootF_trt2 <- read.csv("indic_iG2rootF_trt2.csv")
tax_iG2rootF_trt2 <- left_join(df_iG2rootF_trt2, INRAE_tax, by="OTUID")
write.csv(tax_iG2rootF_trt2, "tax_iG2rootF_trt2.csv")

df_iG2rootF_g1g2 <- read.csv("indic_iG2rootF_g1g2.csv")
tax_iG2rootF_g1g2 <- left_join(df_iG2rootF_g1g2, INRAE_tax, by="OTUID")
write.csv(tax_iG2rootF_g1g2, "tax_iG2rootF_g1g2.csv")

df_iG2rhizoF_trt2 <- read.csv("indic_iG2rhizoF_trt2.csv")
tax_iG2rhizoF_trt2 <- left_join(df_iG2rhizoF_trt2, INRAE_tax, by="OTUID")
write.csv(tax_iG2rhizoF_trt2, "tax_iG2rhizoF_trt2.csv")

df_iG2rhizoF_g1g2 <- read.csv("indic_iG2rhizoF_g1g2.csv")
tax_iG2rhizoF_g1g2 <- left_join(df_iG2rhizoF_g1g2, INRAE_tax, by="OTUID")
write.csv(tax_iG2rhizoF_g1g2, "tax_iG2rhizoF_g1g2.csv")

# Flavert root and rhizo venns
flavert_root_venn <- read.csv("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/inrae_flavert_root_drought_orders.csv", na.strings = "")
iG1rootF_drought <- c(flavert_root_venn$G1_Drought)
iG2rootF_drought <- c(flavert_root_venn$G2_Drought)
iG2rootF_D_D <- c(flavert_root_venn$G2_D_D)

venn.diagram(x = list(iG1rootF_drought, iG2rootF_drought, iG2rootF_D_D), filename="INRAE_F_root_venn.png", imagetype = "png", category.names = c("G1\nDrought", "All G2\nDrought", "G2 Drought_Drought"), fontfamily="sans", cat.fontfamily="sans", main="FR Flavert Drought Root Samples", main.fontfamily = "sans", main.fontface="bold", cat.fontface="bold", disable.logging=TRUE, na="remove", width=5, height=5, units="in", cex=2, main.cex=1.5, cat.cex=1.5, col=c("black", "#3399FF", "#FF6600"), cat.col=c("black", "#3399FF", "#FF6600"))

G1_G2_list_rF <- c(intersect(iG1rootF_drought, iG2rootF_drought))

all_list_rF <- c(intersect(G1_G2_list_rF, iG2rootF_D_D))


# nothing significant in G2 rhizosphere drought so no point making venn diagram, 5 taxa in G1 drought, state this in figure legend


# REDHAWK #

# Generation 1
df_iG1rootRH <- read.csv("indic_iG1rootRH.csv")
tax_iG1rootRH <- left_join(df_iG1rootRH, INRAE_tax, by="OTUID")
write.csv(tax_iG1rootRH, "tax_iG1rootRH.csv")

df_iG1rhizoRH <- read.csv("indic_iG1rhizoRH.csv")
tax_iG1rhizoRH <- left_join(df_iG1rhizoRH, INRAE_tax, by="OTUID")
write.csv(tax_iG1rhizoRH, "tax_iG1rhizoRH.csv")

# Redhawk G2 root have no indicator taxa in G2 drought, 3 taxa in G1 drought, state in figure legend

df_iG2rhizoRH_trt2 <- read.csv("indic_iG2rhizoRH_trt2.csv")
tax_iG2rhizoRH_trt2 <- left_join(df_iG2rhizoRH_trt2, INRAE_tax, by="OTUID")
write.csv(tax_iG2rhizoRH_trt2, "tax_iG2rhizoRH_trt2.csv")

df_iG2rhizoRH_g1g2 <- read.csv("indic_iG2rhizoRH_g1g2.csv")
tax_iG2rhizoRH_g1g2 <- left_join(df_iG2rhizoRH_g1g2, INRAE_tax, by="OTUID")
write.csv(tax_iG2rhizoRH_g1g2, "tax_iG2rhizoRH_g1g2.csv")

# redhawk rhizo venn
redhawk_rhizo_venn <- read.csv("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/inrae_redhawk_rhizo_drought_orders.csv", na.strings = "")
iG1rhizoRH_drought <- c(redhawk_rhizo_venn$G1_Drought)
iG2rhizoRH_drought <- c(redhawk_rhizo_venn$G2_Drought)
iG2rhizoRH_D_D <- c(redhawk_rhizo_venn$G2_D_D)

venn.diagram(x = list(iG1rhizoRH_drought, iG2rhizoRH_drought, iG2rhizoRH_D_D), filename="INRAE_RH_rhizo_venn.png", imagetype = "png", category.names = c("G1\nDrought", "All G2\nDrought", "G2 Drought_\nDrought"), fontfamily="sans", cat.fontfamily="sans", main="FR RedHawk Drought Rhizosphere Samples", main.fontfamily = "sans", main.fontface="bold", cat.fontface="bold", disable.logging=TRUE, na="remove", width=7, height=7, units="in", cex=2.5, main.cex=2, cat.cex=2, col=c("black", "#3399FF", "#FF6600"), cat.col=c("black", "#3399FF", "#FF6600"))

G1_G2_list_zRH <- c(intersect(iG1rhizoRH_drought, iG2rhizoRH_drought))

G1_D_D_zRH <- c(intersect(iG1rhizoRH_drought, iG2rhizoRH_D_D))


```

# Venn diagram?
```{r venn}

### MSU #####
### Roots

MSU_root_venn <- read.csv("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/combined_root_orders.csv", na.strings="")

G1_Control <- c(MSU_root_venn$G1_Control)
G1_Drought <- c(MSU_root_venn$G1_Drought)
G2_Control <- c(MSU_root_venn$G2_Control)
G2_Drought <- c(MSU_root_venn$G2_Drought)
G2_D_D <- c(MSU_root_venn$G2_Drought_Drought)

G1_D_otu <- subset(tax_mG1root, Group == "Drought")
G1_D_otu <- c(G1_D_otu$OTUID)

G2_D_otu <- subset(tax_mG2root_trt2, Group == "Drought")
G2_D_otu <- c(G2_D_otu$OTUID)

# no overlapping ASVs

venn.diagram(x = list(G1_Drought, G2_Drought, G2_D_D), filename="MSU_root_venn.png", imagetype = "png", category.names = c("G1\nDrought", "All G2\nDrought", "G2 Drought_Drought"), fontfamily="sans", cat.fontfamily="sans", main="USA Drought Root Samples", main.fontfamily = "sans", main.fontface="bold", cat.fontface="bold", disable.logging=TRUE, na="remove", width=7, height=7, units="in", cex=2.5, main.cex=2, cat.cex=2, col=c("black", "#3399FF", "#FF6600"), cat.col=c("black", "#3399FF", "#FF6600"))

G1_G2_list <- c(intersect(G1_Drought, G2_Drought))
all_list <- c(intersect(G1_G2_list, G2_D_D))

### Rhizosphere

MSU_rhizo_venn <- read.csv("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/msu_drought_rhizo_orders.csv", na.strings="")
G1_Drought_rhizo <- c(MSU_rhizo_venn$G1_drought)
G2_Drought_rhizo <- c(MSU_rhizo_venn$G2_drought)
G2_D_D_rhizo <- c(MSU_rhizo_venn$G2_Drought_Drought)

venn.diagram(x = list(G1_Drought_rhizo, G2_Drought_rhizo, G2_D_D_rhizo), filename="MSU_rhizo_venn.png", imagetype = "png", category.names = c("G1\nDrought", "All G2\nDrought", "G2 Drought_Drought"), fontfamily="sans", cat.fontfamily="sans", main="USA Drought Rhizosphere Samples", main.fontfamily = "sans", main.fontface="bold", cat.fontface="bold", disable.logging=TRUE, na="none", width=7, height=7, units="in", cex=2.5, main.cex=2, cat.cex=2, col=c("black", "#3399FF", "#FF6600"), cat.col=c("black", "#3399FF", "#FF6600"))

G1_G2_list_z <- c(intersect(G1_Drought_rhizo, G2_Drought_rhizo))

all_list_z <- c(intersect(G1_G2_list_z, G2_D_D_rhizo))
```

# overlapping ASVs
```{r}
#### MSU ####
mG1root_csv <- read.csv(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/tax_mG1root.csv", sep=",")
mG1rhizo_csv <- read.csv(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/tax_mG1rhizo.csv", sep=",")
mG2root_csv <- read.csv(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_taxa_mG2root_full.csv", sep=",")
mG2rhizo_csv <- read.csv(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/indic_taxa_mG2rhizo_full.csv", sep=",")

mG1root_drought <- subset(mG1root_csv, mG1root_csv$Group == "Drought")
mG1rhizo_drought <- subset(mG1rhizo_csv, mG1rhizo_csv$Group == "Drought")
mG2root_g2drought <- subset(mG2root_csv, mG2root_csv$Group == "G2_Drought")
mG2root_DD <- subset(mG2root_csv, mG2root_csv$Group == "Drought_Drought")
mG2rhizo_g2drought <- subset(mG2rhizo_csv, mG2rhizo_csv$Group == "G2_Drought")
mG2rhizo_DD <- subset(mG2rhizo_csv, mG2rhizo_csv$Group == "Drought_Drought")

mG1root_drought_asv <- c(mG1root_drought$OTUID)
mG1rhizo_drought_asv <- c(mG1rhizo_drought$OTUID)
mG2root_g2drought_asv <- c(mG2root_g2drought$OTUID)
mG2root_DD_asv <- c(mG2root_DD$OTUID)
mG2rhizo_g2drought_asv <- c(mG2rhizo_g2drought$OTUID)
mG2rhizo_DD_asv <- c(mG2rhizo_DD$OTUID)
 
root_g1_g2_overlap <- intersect(mG1root_drought_asv, mG2root_g2drought_asv) #2!!! "3a66a38841ba6877568198b9dac3a1dc" "f5c494fd1da6eac3404c69c863c35c07"
root_alldrought <- intersect(root_g1_g2_overlap, mG2root_DD) #none
root_g2_overlap <- intersect(mG2root_DD_asv, mG2root_g2drought_asv) # 1  "4aad5d28a70d09c9aad5df10faf034ac"

rhizo_g1_g2_overlap <- intersect(mG1rhizo_drought_asv, mG2rhizo_g2drought_asv) #none
rhizo_g1_DD_overlap <- intersect(mG1rhizo_drought_asv, mG2rhizo_DD_asv) #none
rhizo_g2_overlap <- intersect(mG2rhizo_DD_asv, mG2rhizo_g2drought_asv) # 3 "6545de8776149403f8ca0bb2479a0ccc" "e27c3062c3ebfd305447410c7c7ca1ea" "c7052a9e9a3f0798ba2598aacd114201"

root_rhizo_G1 <- intersect(mG1rhizo_drought_asv, mG1root_drought_asv) #none
root_rhizo_G2 <- intersect(mG2root_g2drought_asv, mG2rhizo_g2drought_asv) # 2 "47ee012e6bf01ba25c1e9d73e2241846" "7e2b7a47670a5cb9656744205ad8b31f"
root_rhizo_DD <- intersect(mG2root_DD_asv, mG2rhizo_DD_asv) #none


#### France ####
iG1root.rhizoF_csv <- read.csv(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/tax_iG1root.rhizoF_full.csv", sep=",")
iG2rootF_csv <- read.csv(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/tax_iG2rootF_trt2.csv", sep=",")

iG1rootRH_csv <- read.csv(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/tax_iG1rootRH.csv", sep=",")
iG1rhizoRH_csv <- read.csv(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/tax_iG1rhizoRH.csv", sep=",")
iG2rhizoRH_csv <- read.csv(file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/tax_iG2rhizoRH_trt2.csv", sep=",")

iG1F_drought <- subset(iG1root.rhizoF_csv, iG1root.rhizoF_csv$Group == "Drought")
iG1rootF_drought <- subset(iG1F_drought, iG1F_drought$Compartment == "Root")
iG1rhizoF_drought <- subset(iG1F_drought, iG1F_drought$Compartment == "Rhizosphere")
iG2rootF_g2drought <- subset(iG2rootF_csv, iG2rootF_csv$Group == "Drought")

iG1rootRH_drought <- subset(iG1rootRH_csv, iG1rootRH_csv$Group == "G1_Drought")
iG1rhizoRH_drought <- subset(iG1rhizoRH_csv, iG1rhizoRH_csv$Group == "G1_Drought")
iG2rhizoRH_g2drought <- subset(iG2rhizoRH_csv, iG2rhizoRH_csv$Group == "G2_Drought")


#### MSU Pie charts ####
library(ggpubr)

mG1root_pie <- ggpie(data=mG1root_drought, x="Family", label="Family") 
mG1rhizo_drought 
mG2root_g2drought 
mG2rhizo_g2drought

colors_34 <- c("#1F77B4FF", "#AEC7E8FF", "#FF7F0EFF", "#FFBB78FF", "#2CA02CFF", "#98DF8AFF", "#D62728FF", "#FF9896FF", "#9467BDFF", "#C5B0D5FF", "#6B42C8FF", "#8C564BFF", "#C49C94FF", "#E377C2FF", "#C9D73DFF", "#C555CBFF", "#502E71FF", "#C49A3FFF", "#DBDB8DFF", "#D7652DFF", "#CF4C8BFF",  "#17BECFFF", "#9EDAE5FF", "#000000FF", "#5D8D9CFF",  "#C8B693FF", "#FFF9F5FF", "#024663FF", "#70D423FF", "#891604FF", "#FFCE45FF", "#5D734DFF", "#CB7C77FF")

colors_27

colors_17 <- c("#1F77B4FF", "#1F77B4FF", "#9467BDFF", "#1F77B4FF", "#1F77B4FF", "#F7B6D2FF", "#1F77B4FF", "#1F77B4FF", "#9EDAE5FF", "#024663FF", "#1F77B4FF", "#891604FF", "#1F77B4FF", "#000000aa", "#FFCE45FF",  "#5D734DFF", "#CB7C77FF")

mG1root_drought
mG1root_drought_df <- data.frame(table(mG1root_drought$Family))
mG1root_drought_df
unique(mG1root_drought_df$Var1)
mG1root_pie <- ggpie(data=mG1root_drought_df, x="Freq", label = "Var1", lab.pos = "out", lab.font = c(3, "plain", "black"), fill = colors_17) 
mG1root_pie 
ggsave("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/mG1root_pie.png", mG1root_pie, width=6, height = 5, units="in")

mG1rhizo_drought 
mG1rhizo_drought_df <- data.frame(table(mG1rhizo_drought$Family))
mG1rhizo_drought_df
unique(mG1rhizo_drought_df$Var1)
colors_9 <- c("#1F77B4FF", "#D62728FF", "#C5B0D5FF", "#1F77B4FF", "#C49C94FF", "#1F77B4FF", "#000000aa", "#000000aa", "#5D734DFF")
mG1rhizo_pie <- ggpie(data=mG1rhizo_drought_df, x="Freq", label = "Var1", lab.pos = "out", lab.font = c(3, "plain", "black"), fill = colors_9) 
mG1rhizo_pie 
ggsave("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/mG1rhizo_pie.png", mG1rhizo_pie, width=6, height = 5, units="in")

mG2root_g2drought 
mG2root_g2drought_df <- data.frame(table(mG2root_g2drought$Family))
mG2root_g2drought_df
unique(mG2root_g2drought_df$Var1)
colors_25 <- c("#98DF8AFF", "#D62728FF", "#9467BDFF", "#1F77B4FF", "#1F77B4FF", "#1F77B4FF", "#1F77B4FF", "#1F77B4FF", "#BCBD22FF", "#1F77B4FF", "#C49A3FFF", "#1F77B4FF", "#1F77B4FF", "#9EDAE5FF",  "#1F77B4FF", "#1F77B4FF", "#024663FF", "#891604FF", "#1F77B4FF","#1F77B4FF", "#1F77B4FF", "#000000aa", "#1F77B4FF", "#5D734DFF", "#CB7C77FF")
mG2root_g2drought_pie <- ggpie(data=mG2root_g2drought_df, x="Freq", label = "Var1", lab.pos = "out", lab.font = c(3, "plain", "black"), fill = colors_25) 
mG2root_g2drought_pie 
ggsave("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/mG2root_g2drought_pie.png", mG2root_g2drought_pie, width=6, height = 5, units="in")

mG2rhizo_g2drought 
mG2rhizo_g2drought_df <- data.frame(table(mG2rhizo_g2drought$Family))
mG2rhizo_g2drought_df
unique(mG2rhizo_g2drought_df$Var1)
colors_39 <- c("#1F77B4FF", "#AEC7E8FF","#1F77B4FF","#1F77B4FF", "#FFBB78FF","#1F77B4FF","#98DF8AFF", "#9467BDFF", "#1F77B4FF", "#1F77B4FF",
               "#1F77B4FF", "#8C564BFF", "#1F77B4FF", "#1F77B4FF", "#1F77B4FF", "#C9D73DFF", "#1F77B4FF", "#1F77B4FF", "#BCBD22FF", "#1F77B4FF",
               "#1F77B4FF","#1F77B4FF","#1F77B4FF","#1F77B4FF","#1F77B4FF", "#17BECFFF","#1F77B4FF", "#1F77B4FF","#1F77B4FF", "#FFF9F5FF",
               "#1F77B4FF","#024663FF","#1F77B4FF", "#891604FF", "#1F77B4FF","#1F77B4FF","#000000aa","#FFCE45FF","#1F77B4FF")
mG2rhizo_g2drought_pie <- ggpie(data=mG2rhizo_g2drought_df, x="Freq", label = "Var1", lab.pos = "out", lab.font = c(3, "plain", "black"), fill = colors_39) 
mG2rhizo_g2drought_pie 
ggsave("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/mG2rhizo_g2drought_pie.png", mG2rhizo_g2drought_pie, width=6, height = 5, units="in")

msu_RH_pieplots <- (mG1root_pie + mG1rhizo_pie) / (mG2root_g2drought_pie + mG2rhizo_g2drought_pie )
msu_RH_pieplots

#### France Pie Charts ####
iG1rootF_drought 
iG1rhizoF_drought 
iG2rootF_g2drought 

colors_34 <- c("#1F77B4FF", "#AEC7E8FF", "#FF7F0EFF", "#FFBB78FF", "#2CA02CFF", "#98DF8AFF", "#D62728FF", "#FF9896FF", "#9467BDFF", "#C5B0D5FF", "#6B42C8FF", "#8C564BFF", "#C49C94FF", "#E377C2FF", "#C9D73DFF", "#C555CBFF", "#502E71FF", "#C49A3FFF", "#DBDB8DFF", "#D7652DFF", "#CF4C8BFF",  "#17BECFFF", "#9EDAE5FF", "#000000FF", "#5D8D9CFF",  "#C8B693FF", "#FFF9F5FF", "#024663FF", "#70D423FF", "#891604FF", "#FFCE45FF", "#5D734DFF", "#CB7C77FF")

colors_27

iG1rootF_drought 
iG1rootF_drought_df <- data.frame(table(iG1rootF_drought$Family))
iG1rootF_drought_df
unique(iG1rootF_drought_df$Var1)
colors_17_ig1rootF <- c("#FF7F0EFF","#1F77B4FF","#FFBB78FF", "#2CA02CFF", "#D62728FF","#1F77B4FF","#1F77B4FF","#1F77B4FF", "#DBDB8DFF","#D7652DFF",
                        "#17BECFFF", "#1F77B4FF", "#024663FF","#70D423FF","#891604FF","#1F77B4FF", "#CB7C77FF")
iG1rootF_pie <- ggpie(data=iG1rootF_drought_df, x="Freq", label = "Var1", lab.pos = "out", lab.font = c(3, "plain", "black"), fill = colors_17_ig1rootF) 
iG1rootF_pie 
ggsave("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/iG1rootF_pie.png", iG1rootF_pie, width=6, height = 5, units="in")

iG1rhizoF_drought 
iG1rhizoF_drought_df <- data.frame(table(iG1rhizoF_drought$Family))
iG1rhizoF_drought_df
unique(iG1rhizoF_drought_df$Var1)
colors_5_ig1rhizoF <- c("#1F77B4FF","#E377C2FF","#1F77B4FF","#1F77B4FF","#CB7C77FF")
iG1rhizoF_pie <- ggpie(data=iG1rhizoF_drought_df, x="Freq", label = "Var1", lab.pos = "out", lab.font = c(3, "plain", "black"), fill = colors_5_ig1rhizoF) 
iG1rhizoF_pie 
ggsave("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/iG1rhizoF_pie.png", iG1rhizoF_pie, width=6, height = 5, units="in")

iG2rootF_g2drought 
iG2rootF_g2drought_df <- data.frame(table(iG2rootF_g2drought$Family))
iG2rootF_g2drought_df
colors_iG2rootF <- c("#D7652DFF","#891604FF")
iG2rootF_g2drought_pie <- ggpie(data=iG2rootF_g2drought_df, x="Freq", label = "Var1", lab.pos = "out", lab.font = c(3, "plain", "black"), fill = colors_iG2rootF) 
iG2rootF_g2drought_pie 
ggsave("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/iG2rootF_g2drought_pie.png", iG2rootF_g2drought_pie, width=6, height = 5, units="in")


iG1rootRH_drought 
iG1rhizoRH_drought 
iG2rhizoRH_g2drought 

iG1rootRH_drought
iG1rootRH_drought_df <- data.frame(table(iG1rootRH_drought$Family))
iG1rootRH_drought_df
colors_iG1rootRH <- c("#FF7F0EFF", "#DBDB8DFF", "#CB7C77FF")
iG1rootRH_pie <- ggpie(data=iG1rootRH_drought_df, x="Freq", label = "Var1", lab.pos = "out", lab.font = c(3, "plain", "black"), fill = colors_iG1rootRH) 
iG1rootRH_pie 
ggsave("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/iG1rootRH_pie.png", iG1rootRH_pie, width=6, height = 5, units="in")

iG1rhizoRH_drought
iG1rhizoRH_drought_df <- data.frame(table(iG1rhizoRH_drought$Family))
iG1rhizoRH_drought_df
colors_iG1rhizoRH <- c("#AEC7E8FF","#1F77B4FF","#1F77B4FF","#1F77B4FF","#FF7F0EFF","#FFBB78FF","#98DF8AFF", "#6B42C8FF","#8C564BFF","#1F77B4FF",
                       "#1F77B4FF","#1F77B4FF","#1F77B4FF", "#F7B6D2FF","#C555CBFF","#BCBD22FF", "#1F77B4FF","#5D8D9CFF", "#C8B693FF", "#FFF9F5FF",
                        "#024663FF", "#1F77B4FF", "#891604FF","#1F77B4FF","#1F77B4FF", "#5D734DFF", "#CB7C77FF")
iG1rhizoRH_pie <- ggpie(data=iG1rhizoRH_drought_df, x="Freq", label = "Var1", lab.pos = "out", lab.font = c(3, "plain", "black"), fill = colors_iG1rhizoRH) 
iG1rhizoRH_pie 
ggsave("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/iG1rhizoRH_pie.png", iG1rhizoRH_pie, width=6, height = 5, units="in")

iG2rhizoRH_g2drought
iG2rhizoRH_g2drought_df <- data.frame(table(iG2rhizoRH_g2drought$Family))
iG2rhizoRH_g2drought_df
colors_iG2rhizoRH_g2drought <- c("#17BECFFF", "#CB7C77FF")
iG2rhizoRH_g2drought_pie <- ggpie(data=iG2rhizoRH_g2drought_df, x="Freq", label = "Var1", lab.pos = "out", lab.font = c(3, "plain", "black"), fill = colors_iG2rhizoRH_g2drought) 
iG2rhizoRH_g2drought_pie 
ggsave("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/iG2rhizoRH_g2drought_pie.png", iG2rhizoRH_g2drought_pie, width=6, height = 5, units="in")
```

