---
title: "root_OTU_analysis"
author: "Abby Sulesky"
date: "2023-03-09"
output:
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# OTU Analysis for Root samples from multigen experiment

Sequences were demultiplexed, trimmed, merged, and taxonomy assigned in Qiime2 with Silva

Used Figaro to determine where sequences should be trimmed: forward trim 114, reverse trim 161, will merge 90.91% of reads

OTU and taxonomy table were exported as CSV files and will now be analyzed in R

in R do clean up: remove mitochondria, chloroplast and contaminants using decontam package

First change tsv to csv format for taxonomy and OTU tables, make metadata file with negative control samples and true samples

"taxon" column in taxonomy table must be divided into columns for domain, phylum, class etc.
Also need to add "sample_type" column to metadata for negative control decontam later


```{r echo=T, results='hide'}
# load required packages

library(phyloseq)
library(ggplot2)
library(tidyverse)
library(decontam)
library(scales)
library(vegan)

sessionInfo()

#R version 4.2.1 (2022-06-23)
# phyloseq Bioconductor version: Release (3.16) phyloseq version: 1.42.0
# decontam Bioconductor version: Release (3.16) decontam version: 1.18.0

#locale:
#[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

#attached base packages:
#[1] stats     graphics  grDevices utils     datasets  methods   base     

#other attached packages:
# [1] plyr_1.8.8      vegan_2.6-4     lattice_0.20-45 permute_0.9-7   scales_1.2.1    decontam_1.18.0
# [7] forcats_0.5.2   stringr_1.5.0   dplyr_1.0.10    purrr_0.3.5     readr_2.1.3     tidyr_1.2.1    
# [13] tibble_3.1.8    tidyverse_1.3.2 ggplot2_3.4.0   phyloseq_1.42.0
```


## Load Files

```{r echo=T}
setwd("/Users/Abby/OneDrive - Michigan State University/multigen_analysis/roots/")
otu = read.csv("otu_table_multigen_roots.csv", sep=",", row.names=1)
tax = read.csv("taxonomy_multigen_roots.csv", sep=",", row.names=1)
tax = as.matrix(tax)
metadata = read.csv("Root_multigen_metadata.csv", sep=",", row.names=1) 
OTU = otu_table(otu, taxa_are_rows = TRUE)
TAX = tax_table(tax)
meta = sample_data(metadata)

transpose_otu <- t(otu)
write.csv(transpose_otu, "/Users/Abby/OneDrive - Michigan State University/multigen_analysis/roots/transpose_otu_root.csv")
```

## Merge
```{r echo=T}
phyloseq_merged = merge_phyloseq(OTU, meta, TAX)
phyloseq_merged
data.frame(tax_table(phyloseq_merged))

# sample ROOT_G2_3 missing from sequencing


#view_merge <- otu_table(phyloseq_merged)
#class(view_merge) <- "matrix"

```

## remove mitochondria and chloroplast
```{r echo=T}
phyloseq_merged_clean <- phyloseq_merged %>%
  subset_taxa(
      Family   != "f__Chloroplast" &
      Family  != "f__Mitochondria" &
      Phylum != ""
  )
phyloseq_merged_clean

data.frame(tax_table(phyloseq_merged_clean))

#  % taxa that were removed:
percent_taxa_removed <- ((13162-12172)/13162)*100
percent_taxa_removed
```

Removal of mitochondria, chloroplast, and unassigned at "Phylum" removed 7.52% of the total taxa in the samples, leaving 12172 taxa across the 143 samples (including controls)

## inspect library sizes
```{r}
#before host removal
full_merged_df <- as.data.frame(sample_data(phyloseq_merged)) # Put sample_data into a ggplot-friendly data.frame
full_merged_df$LibrarySize <- sample_sums(phyloseq_merged)
full_merged_df <- full_merged_df[order(full_merged_df$LibrarySize),]
full_merged_df$Index <- seq(nrow(full_merged_df))
ggplot(data=full_merged_df, aes(x=Index, y=LibrarySize, color=sample_type)) + geom_point() + ggtitle("Reads before host removal")

df <- as.data.frame(sample_data(phyloseq_merged_clean)) 
df$LibrarySize <- sample_sums(phyloseq_merged_clean)
df <- df[order(df$LibrarySize),]
df$Index <- seq(nrow(df))
ggplot(data=df, aes(x=Index, y=LibrarySize, color=sample_type)) + geom_point() + ggtitle("Reads after host removal")

```

## identify contaminants by prevalence and remove 
in this method, the distribution of the frequency of each sequence feature as a function of the 
prevalence is used to identify contaminants. In this method, the prevalence (presence/absence across samples) of each sequence feature in true positive samples is compared to the prevalence in negative controls to identify contaminants.

```{r}
sample_data(phyloseq_merged_clean)$is.neg <- sample_data(phyloseq_merged_clean)$sample_type == "negative"
contamdf.prev0.1 <- isContaminant(phyloseq_merged_clean, method="prevalence", neg="is.neg")
contamdf.prev0.5 <- isContaminant(phyloseq_merged_clean, method="prevalence", neg="is.neg", threshold=0.5)
table(contamdf.prev0.1$contaminant)
table(contamdf.prev0.5$contaminant)

# FALSE  TRUE 0.1 threshold
# 12166     6 

# FALSE  TRUE 0.5 threshold
# 12150    22

# opted to use the higher contam threshold of 0.5

# Make phyloseq object of presence-absence in negative controls and true samples
ps.pa <- transform_sample_counts(phyloseq_merged_clean, function(abund) 1*(abund>0))
ps.pa.neg <- prune_samples(sample_data(ps.pa)$sample_type == "negative", ps.pa)
ps.pa.pos <- prune_samples(sample_data(ps.pa)$sample_type == "sample", ps.pa)

# Make data.frame of prevalence in positive and negative samples
df.pa <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),
                    contaminant=contamdf.prev0.5$contaminant)


root_0.5_plot <- ggplot(data=df.pa, aes(x=pa.neg, y=pa.pos, color=contaminant)) + geom_point() + labs(x="Prevalence (Negative Controls)", y="Prevalence (True Samples)", title="Root Decontam 0.5")
  
root_0.5_plot

ggsave("Root_0.5_decontam_plot.png", plot=root_0.5_plot,width = 7,height = 5,units = "in")


write.csv(df.pa, "contaminant-table-0.5.csv")
# open contaminant-table in excel and add column name to OTUID, then read back in below

#write.csv(contamdf.prev0.1, "contaminant-prev-0.1.csv")

##removing contaminants from phyloseq object
df.pa <- read.csv("contaminant-table-0.5-edited.csv")
subset.df <- subset(df.pa, contaminant== "FALSE")
keep.taxa <- as.vector(subset.df$OTU)


subset.df.remove <- subset(df.pa, contaminant== "TRUE")
remove.taxa.list <- as.vector(subset.df.remove)

remove.taxa.df <- data.frame(remove.taxa.list)

phyloseq_merged_clean_decontam <- prune_taxa(keep.taxa, phyloseq_merged_clean)
phyloseq_merged_clean_decontam

percent_removed_decontam <- ((12172-12150)/12172)*100
percent_removed_decontam

##filter metadata to true samples only then merge with phyloseq decontam (with removed OTU contaminants)
subset.metadata<- subset(metadata, sample_type =="sample")


keep.samples <- as.vector(subset.metadata$Sample_ID)
keep.samples


phyloseq_merged_clean_decontam_final <- prune_samples(keep.samples, phyloseq_merged_clean_decontam)
phyloseq_merged_clean_decontam_final

## Save phyloseq object for future use
saveRDS(phyloseq_merged_clean_decontam_final, file="/Users/Abby/OneDrive - Michigan State University/multigen_analysis/roots/root_final_phyloseq.rds", compress = TRUE)


```

Removal of contaminants and positive controls removed 0.18% of the total taxa in the samples, leaving 12816 taxa across the 122 true samples

## rarefaction curve
```{r}
decontam_final_otutable <- otu_table(phyloseq_merged_clean_decontam_final)
class(decontam_final_otutable) <- "matrix"

rarecurve <- rarecurve(decontam_final_otutable, step=10, label = FALSE)

# ggsave(rarecurve, file="seed_rainout_rarecurve.tiff", device="tiff", width=3.5, height=3.5, units="in")

```

```{r}
phyloseq::otu_table(phyloseq_merged_clean_decontam_final) %>%
  as.data.frame() %>%
  rownames_to_column("id") %>%
  left_join(phyloseq::tax_table(phyloseq_merged_clean_decontam_final)%>%as.data.frame()%>%
              rownames_to_column("id")) -> otu_decontam_final

write.csv(otu_decontam_final, file = "root_otu_decontam_final.csv")
```





